{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# COMT - Single gene analysis in the NBA GP2 data\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Project: COMT Single Gene analysis\n",
    "- Version: Python/3.10.12; R 4.4.2\n",
    "- Last Updated: 01-APRIL-2025\n",
    "- Update Description: Clean code and adding some minor comments"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Notebook overview\n",
    "- In this notebook we performed association analysis with PLINK using COMT gene variants data in NBA data. We used a loop to perform the same analysis across different GP2 population ancestries.\n",
    "- GP2 NBA Data from Release 7 were used.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Description\n",
    "\n",
    "__1. Description__\n",
    "\n",
    "-  Getting Started\n",
    "- Loading Python libraries\n",
    "- Defining functions\n",
    "- Set Paths\n",
    "- Make working directory\n",
    "\n",
    "__2. Installing packages__\n",
    "\n",
    "__3. Copy over data__\n",
    "\n",
    "__4. Create a covariate file with GP2 data__\n",
    "\n",
    "__5. Remove related individuals and PCA__\n",
    "\n",
    "__6. Annotation of the gene__\n",
    "\n",
    "__7. Case/Control Analyses__\n",
    "\n",
    "__8. Clinical Data Analyses__\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Loading Python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_81/1504764740.py:21: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Defining functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Set paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## GP2 v7.0\n",
    "GP2_RELEASE_PATH = 'path/'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_SUMSTAT_RELEASE_PATH = f'{GP2_RELEASE_PATH}/summary_statistics'\n",
    "\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v7.0')\n",
    "print(f'Path to GP2 v7.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v7.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v7.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Make working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: mkdir -p /home/jupyter/COMT_NBA_loop/\n"
     ]
    }
   ],
   "source": [
    "# Let's create a single folder for all the files\n",
    "WORK_DIR = f'/home/jupyter/COMT_NBA_loop/'\n",
    "shell_do(f'mkdir -p {WORK_DIR}') "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Install Packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## PLINK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink1.9 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink; then\n",
    "echo \"Plink1.9 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink \\n    -------\"\n",
    "wget -N http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20190304.zip \n",
    "unzip -o plink_linux_x86_64_20190304.zip\n",
    "echo -e \"\\n plink downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Install plink2. If this doesn't work, get the updated download link from https://www.cog-genomics.org/plink/2.0/, as this changes regularly. We use the download link for Linux AVX2 AMD."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Plink2 is already installed in /home/jupyter/tools/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p ~/tools\n",
    "cd ~/tools\n",
    "\n",
    "if test -e /home/jupyter/tools/plink2; then\n",
    "echo \"Plink2 is already installed in /home/jupyter/tools/\"\n",
    "\n",
    "else\n",
    "echo -e \"Downloading plink2 \\n    -------\"\n",
    "wget -N https://s3.amazonaws.com/plink2-assets/plink2_linux_amd_avx2_20240806.zip\n",
    "unzip -o plink2_linux_amd_avx2_20240806.zip\n",
    "echo -e \"\\n plink2 downloaded and unzipped in /home/jupyter/tools \\n \"\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%capture\n",
    "%%bash\n",
    "\n",
    "# https://www.openbioinformatics.org/annovar/annovar_download_form.php\n",
    "\n",
    "if test -e /home/jupyter/tools/annovar; then\n",
    "\n",
    "echo \"annovar is already installed in /home/jupyter/tools/\"\n",
    "else\n",
    "echo \"annovar is not installed\"\n",
    "cd /home/jupyter/tools/\n",
    "\n",
    "wget http://www.openbioinformatics.org/annovar/download/0wgxR2rIVP/annovar.latest.tar.gz\n",
    "\n",
    "tar xvfz annovar.latest.tar.gz\n",
    "\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ANNOVAR: Download sources of annotation\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/tools/annovar/\n",
    "\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene humandb/\n",
    "perl annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20170905 humandb/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rvtests is already installed\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Install RVTESTS: Option 1 (~15min)\n",
    "if test -e /home/jupyter/tools/rvtests; then\n",
    "\n",
    "echo \"rvtests is already installed\"\n",
    "else\n",
    "echo \"rvtests is not installed\"\n",
    "\n",
    "mkdir /home/jupyter/tools/rvtests\n",
    "cd /home/jupyter/tools/rvtests\n",
    "\n",
    "wget https://github.com/zhanxw/rvtests/releases/download/v2.1.0/rvtests_linux64.tar.gz \n",
    "\n",
    "tar -zxvf rvtests_linux64.tar.gz\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# chmod to make sure you have permission to run the program\n",
    "! chmod u+x /home/jupyter/tools/plink\n",
    "! chmod u+x /home/jupyter/tools/plink2\n",
    "! chmod 777 /home/jupyter/tools/rvtests/executable/rvtest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annovar\n",
      "annovar.latest.tar.gz\n",
      "LICENSE\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_amd_avx2_20250129.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "rvtests\n",
      "toy.map\n",
      "toy.ped\n",
      "vcf_subset\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#Show installed tools\n",
    "ls /home/jupyter/tools/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Copy Over Files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download clinical data\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_RELEASE_PATH}/clinical_data/master_key_release7_final.csv {WORK_DIR}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_RELEASE_PATH}/clinical_data/extended_clinical_data_release7.csv {WORK_DIR}/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create a covariate file with GP2 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Clinical data (num rows, num columns): (44281, 26)\n"
     ]
    }
   ],
   "source": [
    "# Let's load the master key\n",
    "FULL_PATH = WORK_DIR + f'master_key_release7_final.csv'\n",
    "key = pd.read_csv(FULL_PATH, low_memory=False)\n",
    "print(f'Clinical data (num rows, num columns): {key.shape}')\n",
    "pd.set_option('display.max_columns', None)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subsetting to keep only a few columns \n",
    "key = key[['GP2sampleID', 'baseline_GP2_phenotype_for_qc', 'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset', 'label','GP2ID']]\n",
    "# Renaming the columns\n",
    "key.rename(columns = {'GP2sampleID':'IID',\n",
    "                                     'baseline_GP2_phenotype_for_qc':'phenotype',\n",
    "                                     'biological_sex_for_qc':'SEX', \n",
    "                                     'age_at_sample_collection':'AGE', \n",
    "                                     'age_of_onset':'AAO'}, inplace = True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For each ancestry, we download related samples, and genotyping imputed data.\n",
    "ancestries = {'AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS', 'CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_META_RELEASE_PATH}/related_samples/{ancestry}_release7.related {WORK_DIR}')\n",
    "    shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_RELEASE_PATH}/raw_genotypes/{ancestry}/{ancestry}_release7.eigenvec {WORK_DIR}')\n",
    "    shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_RELEASE_PATH}/raw_genotypes/{ancestry}/{ancestry}_release7* {WORK_DIR}')\n",
    "    shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {GP2_RELEASE_PATH}/imputed_genotypes/{ancestry}/chr22* {WORK_DIR}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestries = {'AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS', 'CAH'}\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    \n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    ## Subset to keep ancestry of interest \n",
    "    ancestry_key = key[key['label']==ancestry].copy()\n",
    "    ancestry_key.reset_index(drop=True)\n",
    "    print(ancestry_key.shape)\n",
    "    # Load information about related individuals in the ancestry analyzed\n",
    "    related_df = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.related')\n",
    "    print(f'Related individuals: {related_df.shape}')\n",
    "\n",
    "    # Make a list of just one set of related people\n",
    "    related_list = list(related_df['IID1'])\n",
    "\n",
    "    # Check value counts of related and remove only one related individual\n",
    "    ancestry_key = ancestry_key[~ancestry_key[\"IID\"].isin(related_list)]\n",
    "\n",
    "    # Check size\n",
    "    print(f'Unrelated individuals: {ancestry_key.shape}')\n",
    "    \n",
    "    # Convert phenotype to binary (1/2)\n",
    "    ## Assign conditions so case=2 and controls=1, and -9 otherwise (matching PLINK convention)\n",
    "    # PD = 2; control = 1\n",
    "    pheno_mapping = {\"PD\": 2, \"Control\": 1}\n",
    "    ancestry_key['PHENO'] = ancestry_key['phenotype'].map(pheno_mapping).astype('Int64')\n",
    "\n",
    "    # Check value counts of pheno\n",
    "    ancestry_key['PHENO'].value_counts(dropna=False)\n",
    "    \n",
    "    ## Get the PCs\n",
    "    pcs = pd.read_csv(f'{WORK_DIR}/{ancestry}_release7.eigenvec', sep='\\t')\n",
    "    \n",
    "    #Select just first 5 PCs\n",
    "    selected_columns = ['IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5']\n",
    "    pcs = pd.DataFrame(data=pcs.iloc[:, 1:7].values, columns=selected_columns)\n",
    "\n",
    "    # Reset the index to remove any potential issues\n",
    "    pcs = pcs.reset_index(drop=True)\n",
    "    \n",
    "    # Check size\n",
    "    print(f'PCs: {pcs.shape}')\n",
    "\n",
    "    # Check value counts of SEX\n",
    "    sex_og_values = ancestry_key['SEX'].value_counts(dropna=False)\n",
    "    print(f'Sex value counts - original:\\n {sex_og_values.to_string()}')\n",
    "    \n",
    "    # Convert sex to binary (1/2)\n",
    "    ## Assign conditions so female=2 and men=1, and -9 otherwise (matching PLINK convention)\n",
    "    # Female = 2; Male = 1\n",
    "    sex_mapping = {\"Female\": 2, \"Male\": 1}\n",
    "    ancestry_key['SEX'] = ancestry_key['SEX'].map(sex_mapping).astype('Int64')\n",
    "    \n",
    "    # Check value counts of SEX after recoding\n",
    "    sex_recode_values = ancestry_key['SEX'].value_counts(dropna=False)\n",
    "    print(f'Sex value counts - recoded:\\n{sex_recode_values.to_string()}')\n",
    "    \n",
    "    ## Make covariate file\n",
    "    df = pd.merge(pcs, ancestry_key, on='IID')\n",
    "    print(f'Check columns for covariate file: {df.columns}')\n",
    "    \n",
    "    #Make additional columns - FID, fatid and matid - these are needed for RVtests!!\n",
    "    #RVtests needs the first 5 columns to be fid, iid, fatid, matid and sex otherwise it does not run correctly\n",
    "    #Uppercase column name is ok\n",
    "    #See https://zhanxw.github.io/rvtests/#phenotype-file\n",
    "    df['FID'] = 0\n",
    "    df['FATID'] = 0\n",
    "    df['MATID'] = 0\n",
    "\n",
    "    ## Clean up and keep columns we need \n",
    "    final_df = df[['FID','IID', 'FATID', 'MATID', 'SEX', 'AGE', 'PHENO','PC1', 'PC2', 'PC3', 'PC4', 'PC5','GP2ID']].copy()\n",
    "\n",
    "    ##DO NOT replace missing values with -9 as this is misinterpreted by RVtests - needs to be nonnumeric\n",
    "    #Leave missing values as NA\n",
    "    \n",
    "    #Check number of PD cases missing age\n",
    "    pd_missAge = final_df[(final_df['PHENO']==2)&(final_df['AGE'].isna())]\n",
    "    print(f'Number of PD cases missing age: {pd_missAge.shape[0]}')\n",
    "    \n",
    "    #Check number of controls missing age\n",
    "    control_missAge = final_df[(final_df['PHENO']==1)&(final_df['AGE'].isna())]\n",
    "    print(f'Number of controls missing age: {control_missAge.shape[0]}')\n",
    "\n",
    "    ## Make file of sample IDs to keep \n",
    "    samples_toKeep = final_df[['FID', 'IID']].copy()\n",
    "    samples_toKeep.to_csv(f'{WORK_DIR}/samplestokeep_{ancestry}.txt', sep=\"\\t\",index=None)\n",
    "    \n",
    "    final_df.to_csv(f'{WORK_DIR}/{ancestry}_covariate_file.txt', sep = '\\t', na_rep='NA', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotation of the gene\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- COMT coordinates: chr22: 19,941,371-19,969,975 (GRCh38/hg38) (forward strand - ensemble)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extract the region using PLINK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AMR_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AMR.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//AMR_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_AMR_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:45:58 2025\n",
      "7445 MiB RAM detected, ~5619 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "614 samples (271 females, 343 males; 614 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_AMR_release7.psam.\n",
      "465193 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_AMR_release7.pvar.\n",
      "1 binary phenotype loaded (431 cases, 150 controls).\n",
      "--keep: 609 samples remaining.\n",
      "609 samples (268 females, 341 males; 609 founders) remaining after main\n",
      "filters.\n",
      "427 cases and 150 controls remaining after main filters.\n",
      "Calculating allele frequencies... done.\n",
      "191 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "249 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AMR_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AMR_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AMR_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:45:58 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//MDE_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_MDE.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//MDE_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_MDE_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:45:59 2025\n",
      "7445 MiB RAM detected, ~5606 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "470 samples (173 females, 297 males; 470 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_MDE_release7.psam.\n",
      "402408 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_MDE_release7.pvar.\n",
      "1 binary phenotype loaded (234 cases, 204 controls).\n",
      "--keep: 457 samples remaining.\n",
      "457 samples (166 females, 291 males; 457 founders) remaining after main\n",
      "filters.\n",
      "230 cases and 196 controls remaining after main filters.\n",
      "Calculating allele frequencies... done.\n",
      "214 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "203 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//MDE_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//MDE_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//MDE_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:45:59 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AFR_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AFR.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//AFR_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_AFR_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:46:00 2025\n",
      "7445 MiB RAM detected, ~5612 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "2632 samples (1159 females, 1473 males; 2632 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_AFR_release7.psam.\n",
      "870655 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_AFR_release7.pvar.\n",
      "1 binary phenotype loaded (933 cases, 1677 controls).\n",
      "--keep: 2567 samples remaining.\n",
      "2567 samples (1125 females, 1442 males; 2567 founders) remaining after main\n",
      "filters.\n",
      "929 cases and 1617 controls remaining after main filters.\n",
      "Calculating allele frequencies... done.\n",
      "278 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "487 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AFR_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AFR_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AFR_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:46:02 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EAS_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EAS.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//EAS_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_EAS_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:46:03 2025\n",
      "7445 MiB RAM detected, ~5606 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "5136 samples (1868 females, 3268 males; 5136 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_EAS_release7.psam.\n",
      "645948 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_EAS_release7.pvar.\n",
      "1 binary phenotype loaded (2644 cases, 2452 controls).\n",
      "--keep: 4932 samples remaining.\n",
      "4932 samples (1804 females, 3128 males; 4932 founders) remaining after main\n",
      "filters.\n",
      "2547 cases and 2345 controls remaining after main filters.\n",
      "Calculating allele frequencies... done.\n",
      "193 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "420 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//EAS_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//EAS_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//EAS_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:46:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//SAS_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_SAS.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//SAS_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_SAS_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:46:05 2025\n",
      "7445 MiB RAM detected, ~5615 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "576 samples (200 females, 376 males; 576 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_SAS_release7.psam.\n",
      "377260 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_SAS_release7.pvar.\n",
      "1 binary phenotype loaded (338 cases, 216 controls).\n",
      "--keep: 539 samples remaining.\n",
      "539 samples (188 females, 351 males; 539 founders) remaining after main\n",
      "filters.\n",
      "319 cases and 198 controls remaining after main filters.\n",
      "Calculating allele frequencies... done.\n",
      "188 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "203 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//SAS_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//SAS_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//SAS_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:46:06 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//CAH_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_CAH.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//CAH_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_CAH_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:46:07 2025\n",
      "7445 MiB RAM detected, ~5604 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "823 samples (393 females, 430 males; 823 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_CAH_release7.psam.\n",
      "691299 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_CAH_release7.pvar.\n",
      "1 binary phenotype loaded (503 cases, 297 controls).\n",
      "--keep: 800 samples remaining.\n",
      "800 samples (378 females, 422 males; 800 founders) remaining after main\n",
      "filters.\n",
      "494 cases and 283 controls remaining after main filters.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... done.\n",
      "258 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "366 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//CAH_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//CAH_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//CAH_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:46:09 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AAC_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AAC.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//AAC_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_AAC_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:46:10 2025\n",
      "7445 MiB RAM detected, ~5609 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "1105 samples (650 females, 455 males; 1105 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_AAC_release7.psam.\n",
      "755699 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_AAC_release7.pvar.\n",
      "1 binary phenotype loaded (281 cases, 799 controls).\n",
      "--keep: 1097 samples remaining.\n",
      "1097 samples (647 females, 450 males; 1097 founders) remaining after main\n",
      "filters.\n",
      "279 cases and 793 controls remaining after main filters.\n",
      "Calculating allele frequencies... done.\n",
      "301 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "413 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AAC_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AAC_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AAC_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:46:11 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EUR_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EUR.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//EUR_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_EUR_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:46:12 2025\n",
      "7445 MiB RAM detected, ~5630 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "25795 samples (10129 females, 15666 males; 25795 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_EUR_release7.psam.\n",
      "1970326 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_EUR_release7.pvar.\n",
      "1 binary phenotype loaded (13260 cases, 5295 controls).\n",
      "--keep: 25309 samples remaining.\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) remaining after main\n",
      "filters.\n",
      "13034 cases and 5191 controls remaining after main filters.\n",
      "Calculating allele frequencies... done.\n",
      "916 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "994 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//EUR_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//EUR_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//EUR_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:46:16 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AJ_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AJ.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//AJ_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_AJ_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:46:17 2025\n",
      "7445 MiB RAM detected, ~5616 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "2577 samples (987 females, 1590 males; 2577 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_AJ_release7.psam.\n",
      "321663 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_AJ_release7.pvar.\n",
      "1 binary phenotype loaded (1246 cases, 393 controls).\n",
      "--keep: 2454 samples remaining.\n",
      "2454 samples (915 females, 1539 males; 2454 founders) remaining after main\n",
      "filters.\n",
      "1234 cases and 391 controls remaining after main filters.\n",
      "Calculating allele frequencies... done.\n",
      "71 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "245 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AJ_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AJ_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//AJ_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:46:18 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//CAS_COMT.log.\n",
      "Options in effect:\n",
      "  --chr 22\n",
      "  --from-bp 19941371\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_CAS.txt\n",
      "  --mac 2\n",
      "  --make-bed\n",
      "  --max-alleles 2\n",
      "  --out /home/jupyter/COMT_NBA_loop//CAS_COMT\n",
      "  --pfile /home/jupyter/COMT_NBA_loop//chr22_CAS_release7\n",
      "  --to-bp 19969975\n",
      "\n",
      "Start time: Fri Mar 14 16:46:19 2025\n",
      "7445 MiB RAM detected, ~5636 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "889 samples (461 females, 428 males; 889 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_CAS_release7.psam.\n",
      "424191 variants loaded from\n",
      "/home/jupyter/COMT_NBA_loop//chr22_CAS_release7.pvar.\n",
      "1 binary phenotype loaded (541 cases, 341 controls).\n",
      "--keep: 831 samples remaining.\n",
      "831 samples (439 females, 392 males; 831 founders) remaining after main\n",
      "filters.\n",
      "519 cases and 306 controls remaining after main filters.\n",
      "Calculating allele frequencies... done.\n",
      "207 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "229 variants remaining after main filters.\n",
      "Writing /home/jupyter/COMT_NBA_loop//CAS_COMT.fam ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//CAS_COMT.bim ... done.\n",
      "Writing /home/jupyter/COMT_NBA_loop//CAS_COMT.bed ... done.\n",
      "End time: Fri Mar 14 16:46:20 2025\n"
     ]
    }
   ],
   "source": [
    "## extract region using plink for each ancestry\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    ! /home/jupyter/plink2 \\\n",
    "    --pfile {WORK_DIR}/chr22_{ancestry}_release7 \\\n",
    "    --chr 22 \\\n",
    "    --from-bp 19941371 \\\n",
    "    --to-bp 19969975 \\\n",
    "    --keep {WORK_DIR}/samplestokeep_{ancestry}.txt \\\n",
    "    --max-alleles 2 \\\n",
    "    --mac 2 \\\n",
    "    --make-bed \\\n",
    "    --out {WORK_DIR}/{ancestry}_COMT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Turn binary files into VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AMR_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AMR_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//AMR_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:21 2025\n",
      "7445 MiB RAM detected, ~5613 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "609 samples (268 females, 341 males; 609 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//AMR_COMT.fam.\n",
      "249 variants loaded from /home/jupyter/COMT_NBA_loop//AMR_COMT.bim.\n",
      "1 binary phenotype loaded (427 cases, 150 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//AMR_COMT.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:21 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//MDE_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//MDE_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//MDE_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:22 2025\n",
      "7445 MiB RAM detected, ~5613 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "457 samples (166 females, 291 males; 457 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//MDE_COMT.fam.\n",
      "203 variants loaded from /home/jupyter/COMT_NBA_loop//MDE_COMT.bim.\n",
      "1 binary phenotype loaded (230 cases, 196 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//MDE_COMT.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:22 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AFR_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AFR_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//AFR_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:23 2025\n",
      "7445 MiB RAM detected, ~5612 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "2567 samples (1125 females, 1442 males; 2567 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//AFR_COMT.fam.\n",
      "487 variants loaded from /home/jupyter/COMT_NBA_loop//AFR_COMT.bim.\n",
      "1 binary phenotype loaded (929 cases, 1617 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//AFR_COMT.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:23 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EAS_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EAS_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//EAS_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:25 2025\n",
      "7445 MiB RAM detected, ~5614 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "4932 samples (1804 females, 3128 males; 4932 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//EAS_COMT.fam.\n",
      "420 variants loaded from /home/jupyter/COMT_NBA_loop//EAS_COMT.bim.\n",
      "1 binary phenotype loaded (2547 cases, 2345 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//EAS_COMT.vcf ... 101011111212131314151516161717181819202021212222232324252526262727282829303031313232333334353536363737383839404041414242434344454546464747484849505051515252535354555556565757585859606061616262636364656566666767686869707071717272737374757576767777787879808081818282838384858586868787888889909091919292939394959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:25 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//SAS_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//SAS_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//SAS_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:26 2025\n",
      "7445 MiB RAM detected, ~5622 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "539 samples (188 females, 351 males; 539 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//SAS_COMT.fam.\n",
      "203 variants loaded from /home/jupyter/COMT_NBA_loop//SAS_COMT.bim.\n",
      "1 binary phenotype loaded (319 cases, 198 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//SAS_COMT.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:26 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//CAH_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//CAH_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//CAH_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:27 2025\n",
      "7445 MiB RAM detected, ~5622 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "800 samples (378 females, 422 males; 800 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//CAH_COMT.fam.\n",
      "366 variants loaded from /home/jupyter/COMT_NBA_loop//CAH_COMT.bim.\n",
      "1 binary phenotype loaded (494 cases, 283 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//CAH_COMT.vcf ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:27 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AAC_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AAC_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//AAC_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:28 2025\n",
      "7445 MiB RAM detected, ~5623 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "1097 samples (647 females, 450 males; 1097 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//AAC_COMT.fam.\n",
      "413 variants loaded from /home/jupyter/COMT_NBA_loop//AAC_COMT.bim.\n",
      "1 binary phenotype loaded (279 cases, 793 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//AAC_COMT.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:28 2025\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EUR_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EUR_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//EUR_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:29 2025\n",
      "7445 MiB RAM detected, ~5621 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//EUR_COMT.fam.\n",
      "994 variants loaded from /home/jupyter/COMT_NBA_loop//EUR_COMT.bim.\n",
      "1 binary phenotype loaded (13034 cases, 5191 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//EUR_COMT.vcf ... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:29 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AJ_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AJ_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//AJ_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:30 2025\n",
      "7445 MiB RAM detected, ~5607 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "2454 samples (915 females, 1539 males; 2454 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//AJ_COMT.fam.\n",
      "245 variants loaded from /home/jupyter/COMT_NBA_loop//AJ_COMT.bim.\n",
      "1 binary phenotype loaded (1234 cases, 391 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//AJ_COMT.vcf ... 10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:30 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//CAS_COMT.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//CAS_COMT\n",
      "  --export vcf id-paste=iid\n",
      "  --out /home/jupyter/COMT_NBA_loop//CAS_COMT\n",
      "\n",
      "Start time: Fri Mar 14 16:46:31 2025\n",
      "7445 MiB RAM detected, ~5605 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "831 samples (439 females, 392 males; 831 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//CAS_COMT.fam.\n",
      "229 variants loaded from /home/jupyter/COMT_NBA_loop//CAS_COMT.bim.\n",
      "1 binary phenotype loaded (519 cases, 306 controls).\n",
      "--export vcf to /home/jupyter/COMT_NBA_loop//CAS_COMT.vcf ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Fri Mar 14 16:46:31 2025\n"
     ]
    }
   ],
   "source": [
    "for ancestry in ancestries:\n",
    "        \n",
    "    ## Turn binary files into VCF\n",
    "    ! /home/jupyter/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_COMT \\\n",
    "    --recode vcf id-paste=iid \\\n",
    "    --out {WORK_DIR}/{ancestry}_COMT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Bgzip and Tabix (zip and index the file)\n",
    "for ancestry in ancestries:    \n",
    "    ! bgzip -f {WORK_DIR}/{ancestry}_COMT.vcf\n",
    "    ! tabix -f -p vcf {WORK_DIR}/{ancestry}_COMT.vcf.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate using ANNOVAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for ancestry in ancestries:\n",
    "        \n",
    "    ! perl /home/jupyter/tools/annovar/table_annovar.pl {WORK_DIR}/{ancestry}_COMT.vcf.gz /home/jupyter/tools/annovar/humandb/ -buildver hg38 \\\n",
    "    -out {WORK_DIR}/{ancestry}_COMT.annovar \\\n",
    "    -remove -protocol refGene,clinvar_20170905 \\\n",
    "    -operation g,f \\\n",
    "    --nopolish \\\n",
    "    -nastring . \\\n",
    "    -vcfinput\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "        \n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    # Read in ANNOVAR multianno file\n",
    "    gene = pd.read_csv(f'{WORK_DIR}/{ancestry}_COMT.annovar.hg38_multianno.txt', sep = '\\t')\n",
    "    \n",
    "    #Filter for the correct gene name (sometimes other genes are also included)\n",
    "    gene = gene[gene['Gene.refGene'] == 'COMT']\n",
    "    \n",
    "    #Print number of variants in the different categories\n",
    "    results = [] \n",
    "\n",
    "    intronic = gene[gene['Func.refGene']== 'intronic']\n",
    "    upstream = gene[gene['Func.refGene']== 'upstream']\n",
    "    downstream = gene[gene['Func.refGene']== 'downstream']\n",
    "    utr5 = gene[gene['Func.refGene']== 'UTR5']\n",
    "    utr3 = gene[gene['Func.refGene']== 'UTR3']\n",
    "    splicing = gene[gene['Func.refGene']== 'splicing']\n",
    "    exonic = gene[gene['Func.refGene']== 'exonic']\n",
    "    stopgain = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'stopgain')]\n",
    "    stoploss = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'stoploss')]\n",
    "    startloss = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'startloss')]\n",
    "    frameshift_deletion = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'frameshift deletion')]\n",
    "    frameshift_insertion = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'frameshift insertion')]\n",
    "    nonframeshift_deletion = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonframeshift deletion')]\n",
    "    nonframeshift_insertion = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonframeshift insertion')]\n",
    "    coding_nonsynonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'nonsynonymous SNV')]\n",
    "    coding_synonymous = gene[(gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] == 'synonymous SNV')]\n",
    "        \n",
    "    print({ancestry})\n",
    "    print('Total variants: ', len(gene))\n",
    "    print(\"Intronic: \", len(intronic))\n",
    "    print(\"Upstream: \", len(upstream))\n",
    "    print(\"Downstream: \", len(downstream))\n",
    "    print('UTR3: ', len(utr3))\n",
    "    print('UTR5: ', len(utr5))\n",
    "    print(\"Splicing: \", len(splicing))\n",
    "    print(\"Total exonic: \", len(exonic))\n",
    "    print(\"Stopgain: \", len(stopgain))\n",
    "    print(\"Stoploss: \", len(stoploss))\n",
    "    print(\"Startloss: \", len(startloss))\n",
    "    print(\"Frameshift deletion: \", len(frameshift_deletion))\n",
    "    print(\"Frameshift insertion: \", len(frameshift_insertion))\n",
    "    print(\"Non-frameshift insertion: \", len(nonframeshift_insertion))\n",
    "    print(\"Non-frameshift deletion: \", len(nonframeshift_deletion))\n",
    "    print('Synonymous: ', len(coding_synonymous))\n",
    "    print(\"Nonsynonymous: \", len(coding_nonsynonymous))\n",
    "    results.append((gene, intronic, upstream, downstream, utr3, utr5, splicing,exonic,stopgain,stoploss,startloss, frameshift_deletion,frameshift_insertion,nonframeshift_deletion,nonframeshift_insertion,coding_synonymous, coding_nonsynonymous))\n",
    "    print('\\n')\n",
    "    \n",
    "    ## For rvtests\n",
    "    \n",
    "    # Potential functional: These are variants annotated as frameshift, nonframeshift, startloss, stoploss, stopgain, splicing, missense, exonic, UTR5, UTR3, upstream (-100bp), downstream (+100bp), or ncRNA. \n",
    "    potentially_functional = gene[gene['Func.refGene'] != 'intronic']\n",
    "    # Coding: These are variants annotated as frameshift, nonframeshift, startloss, stoploss, stopgain, splicing, or missense.\n",
    "    coding_variants = gene[(gene['Func.refGene'] == 'splicing') | (gene['Func.refGene'] == 'exonic') & (gene['ExonicFunc.refGene'] != 'synonymous SNV')]\n",
    "    # Loss of function: These are variants annotated as frameshift, startloss,stopgain, or splicing.\n",
    "    loss_of_function = gene[(gene['Func.refGene'] == 'splicing') | (gene['ExonicFunc.refGene'] == 'stopgain') | (gene['ExonicFunc.refGene'] == 'startloss') | (gene['ExonicFunc.refGene'] == 'frameshift deletion') | (gene['ExonicFunc.refGene'] == 'frameshift insertion')]\n",
    "    \n",
    "    # Save in PLINK format\n",
    "    variants_toKeep = potentially_functional[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep.to_csv(f'{WORK_DIR}/{ancestry}_COMT.potentially_functional.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "\n",
    "    variants_toKeep2 = coding_variants[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep2.to_csv(f'{WORK_DIR}/{ancestry}_COMT.coding_variants.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "\n",
    "    variants_toKeep3 = loss_of_function[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep3.to_csv(f'{WORK_DIR}/{ancestry}_COMT.loss_of_function.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)\n",
    "    \n",
    "    # These are all exonic variants\n",
    "    exonic = gene[gene['Func.refGene'] == 'exonic']\n",
    "    \n",
    "    # Save in PLINK format\n",
    "    variants_toKeep4 = exonic[['Chr', 'Start', 'End', 'Gene.refGene']].copy()\n",
    "    variants_toKeep4.to_csv(f'{WORK_DIR}/{ancestry}_COMT.exonic.variantstoKeep.txt', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Burden Analyses using RVTests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['potentially_functional', 'coding_variants','loss_of_function']\n",
    "\n",
    "#Loop over all the ancestries and the 3 variant classes\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "                \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running plink to extract {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        #Extract relevant variants\n",
    "        ! /home/jupyter/plink2 \\\n",
    "        --pfile {WORK_DIR}/chr22_{ancestry}_release7 \\\n",
    "        --keep {WORK_DIR}/samplestokeep_{ancestry}.txt \\\n",
    "        --extract range {WORK_DIR}/{ancestry}_COMT.{variant_class}.variantstoKeep.txt \\\n",
    "        --recode vcf-iid \\\n",
    "        --out {WORK_DIR}/{ancestry}_COMT.{variant_class}\n",
    "\n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running bgzip and tabix for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## Bgzip and Tabix (zip and index the file)\n",
    "        ! bgzip -f {WORK_DIR}/{ancestry}_COMT.{variant_class}.vcf\n",
    "        ! tabix -f -p vcf {WORK_DIR}/{ancestry}_COMT.{variant_class}.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORKSPACE_BUCKET}/refFlat.txt.gz {WORK_DIR}')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "!gzip -d -f {WORK_DIR}/refFlat.txt.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Run RVtests\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS', 'CAH']\n",
    "variant_classes = ['potentially_functional', 'coding_variants','loss_of_function']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "    for variant_class in variant_classes:\n",
    "                \n",
    "        # Print the command to be executed (for debugging purposes)\n",
    "        print(f'Running RVtests for {variant_class} variants for ancestry: {ancestry}')\n",
    "        \n",
    "        ## RVtests with covariates \n",
    "        #Make sure the pheno and covariate file starts with the first 5 columsn: fid, iid, fatid, matid, sex\n",
    "        #The pheno-name flag only works when the pheno/covar file is structured properly\n",
    "        ! /home/jupyter/tools/rvtests/executable/rvtest --noweb --hide-covar \\\n",
    "        --out {WORK_DIR}/{ancestry}_COMT.burden.{variant_class} \\\n",
    "        --kernel skat,skato \\\n",
    "        --inVcf {WORK_DIR}/{ancestry}_COMT.{variant_class}.vcf.gz \\\n",
    "        --pheno {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "        --pheno-name PHENO \\\n",
    "        --gene COMT \\\n",
    "        --geneFile {WORK_DIR}/refFlat.txt \\\n",
    "        --covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "        --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Removed FIN and MDE from ancestry_GROUPS as there are no FIN samples and no variants of interest in MDE and CAH (only 4 samples)\n",
    "ancestry_GROUPS = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'SAS']\n",
    "#ancestry_GROUPS = ['CAH','MDE']\n",
    "variant_classes = ['potentially_functional', 'coding_variants', 'loss_of_function']\n",
    "\n",
    "for ancestry_GROUP in ancestry_GROUPS:\n",
    "    for variant_class in variant_classes:\n",
    "        \n",
    "        shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}_COMT.burden.{variant_class}.*.assoc {WORKSPACE_BUCKET}/COMT_NBA_loop/Burden/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Case/Control Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "__1. Glossary__\n",
    "- CHR Chromosome code\n",
    "- SNP Variant identifier\n",
    "- A1 Allele 1 (usually minor)\n",
    "- A2 Allele 2 (usually major)\n",
    "- MAF Allele 1 frequency in all subjects\n",
    "- F_A/MAF_A Allele 1 frequency in cases\n",
    "- F_U/MAF_U Allele 1 frequency in controls\n",
    "- NCHROBS_A Number of case allele observations\n",
    "- NCHROBS_U Number of control allele observations\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## All variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AAC_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AAC_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AAC.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//AAC_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "413 variants loaded from .bim file.\n",
      "1097 people (450 males, 647 females) loaded from .fam.\n",
      "1072 phenotype values loaded from .fam.\n",
      "--keep: 1097 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1097 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997596.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "227 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "186 variants and 1097 people pass filters and QC.\n",
      "Among remaining phenotypes, 279 are cases and 793 are controls.  (25 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//AAC_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 2.03468.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (186 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//AAC_COMT.AllVariants.assoc.adjusted .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AAC_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AAC_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AAC.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//AAC_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "413 variants loaded from .bim file.\n",
      "1097 people (450 males, 647 females) loaded from .fam.\n",
      "1072 phenotype values loaded from .fam.\n",
      "--keep: 1097 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1097 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997596.\n",
      "413 variants and 1097 people pass filters and QC.\n",
      "Among remaining phenotypes, 279 are cases and 793 are controls.  (25 phenotypes\n",
      "are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//AAC_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AFR_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AFR_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AFR.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//AFR_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "487 variants loaded from .bim file.\n",
      "2567 people (1442 males, 1125 females) loaded from .fam.\n",
      "2546 phenotype values loaded from .fam.\n",
      "--keep: 2567 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2567 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99684.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "306 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "181 variants and 2567 people pass filters and QC.\n",
      "Among remaining phenotypes, 929 are cases and 1617 are controls.  (21\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//AFR_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.37832.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (181 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//AFR_COMT.AllVariants.assoc.adjusted .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AFR_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AFR_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AFR.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//AFR_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "487 variants loaded from .bim file.\n",
      "2567 people (1442 males, 1125 females) loaded from .fam.\n",
      "2546 phenotype values loaded from .fam.\n",
      "--keep: 2567 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2567 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99684.\n",
      "487 variants and 2567 people pass filters and QC.\n",
      "Among remaining phenotypes, 929 are cases and 1617 are controls.  (21\n",
      "phenotypes are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//AFR_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AJ_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AJ_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AJ.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//AJ_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "245 variants loaded from .bim file.\n",
      "2454 people (1539 males, 915 females) loaded from .fam.\n",
      "1625 phenotype values loaded from .fam.\n",
      "--keep: 2454 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2454 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999039.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "117 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "128 variants and 2454 people pass filters and QC.\n",
      "Among remaining phenotypes, 1234 are cases and 391 are controls.  (829\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//AJ_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (128 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//AJ_COMT.AllVariants.assoc.adjusted .\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AJ_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AJ_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AJ.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//AJ_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "245 variants loaded from .bim file.\n",
      "2454 people (1539 males, 915 females) loaded from .fam.\n",
      "1625 phenotype values loaded from .fam.\n",
      "--keep: 2454 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2454 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999039.\n",
      "245 variants and 2454 people pass filters and QC.\n",
      "Among remaining phenotypes, 1234 are cases and 391 are controls.  (829\n",
      "phenotypes are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//AJ_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AMR_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AMR_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AMR.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//AMR_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "249 variants loaded from .bim file.\n",
      "609 people (341 males, 268 females) loaded from .fam.\n",
      "577 phenotype values loaded from .fam.\n",
      "--keep: 609 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 609 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99516.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "125 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "124 variants and 609 people pass filters and QC.\n",
      "Among remaining phenotypes, 427 are cases and 150 are controls.  (32 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//AMR_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.93543.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (124 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//AMR_COMT.AllVariants.assoc.adjusted .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//AMR_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//AMR_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_AMR.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//AMR_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "249 variants loaded from .bim file.\n",
      "609 people (341 males, 268 females) loaded from .fam.\n",
      "577 phenotype values loaded from .fam.\n",
      "--keep: 609 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 609 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99516.\n",
      "249 variants and 609 people pass filters and QC.\n",
      "Among remaining phenotypes, 427 are cases and 150 are controls.  (32 phenotypes\n",
      "are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//AMR_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//CAS_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//CAS_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_CAS.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//CAS_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "229 variants loaded from .bim file.\n",
      "831 people (392 males, 439 females) loaded from .fam.\n",
      "825 phenotype values loaded from .fam.\n",
      "--keep: 831 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 831 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.988455.\n",
      "Warning: --hwe observation counts vary by more than 10%.  Consider using\n",
      "--geno, and/or applying different p-value thresholds to distinct subsets of\n",
      "your data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "108 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "121 variants and 831 people pass filters and QC.\n",
      "Among remaining phenotypes, 519 are cases and 306 are controls.  (6 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//CAS_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 3.07418.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (121 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//CAS_COMT.AllVariants.assoc.adjusted .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//CAS_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//CAS_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_CAS.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//CAS_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "229 variants loaded from .bim file.\n",
      "831 people (392 males, 439 females) loaded from .fam.\n",
      "825 phenotype values loaded from .fam.\n",
      "--keep: 831 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 831 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.988455.\n",
      "229 variants and 831 people pass filters and QC.\n",
      "Among remaining phenotypes, 519 are cases and 306 are controls.  (6 phenotypes\n",
      "are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//CAS_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EAS_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EAS_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EAS.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//EAS_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "420 variants loaded from .bim file.\n",
      "4932 people (3128 males, 1804 females) loaded from .fam.\n",
      "4892 phenotype values loaded from .fam.\n",
      "--keep: 4932 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 4932 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.988089.\n",
      "Warning: --hwe observation counts vary by more than 10%.  Consider using\n",
      "--geno, and/or applying different p-value thresholds to distinct subsets of\n",
      "your data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "305 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "115 variants and 4932 people pass filters and QC.\n",
      "Among remaining phenotypes, 2547 are cases and 2345 are controls.  (40\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//EAS_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 7.01227.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (115 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//EAS_COMT.AllVariants.assoc.adjusted .\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EAS_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EAS_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EAS.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//EAS_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "420 variants loaded from .bim file.\n",
      "4932 people (3128 males, 1804 females) loaded from .fam.\n",
      "4892 phenotype values loaded from .fam.\n",
      "--keep: 4932 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 4932 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.988089.\n",
      "420 variants and 4932 people pass filters and QC.\n",
      "Among remaining phenotypes, 2547 are cases and 2345 are controls.  (40\n",
      "phenotypes are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//EAS_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EUR_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EUR.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "994 variants loaded from .bim file.\n",
      "25309 people (15423 males, 9886 females) loaded from .fam.\n",
      "18225 phenotype values loaded from .fam.\n",
      "--keep: 25309 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 25309 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.933442.\n",
      "Warning: --hwe observation counts vary by more than 10%.  Consider using\n",
      "--geno, and/or applying different p-value thresholds to distinct subsets of\n",
      "your data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "877 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "117 variants and 25309 people pass filters and QC.\n",
      "Among remaining phenotypes, 13034 are cases and 5191 are controls.  (7084\n",
      "phenotypes are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.29886.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (117 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants.assoc.adjusted .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EUR_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EUR.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "994 variants loaded from .bim file.\n",
      "25309 people (15423 males, 9886 females) loaded from .fam.\n",
      "18225 phenotype values loaded from .fam.\n",
      "--keep: 25309 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 25309 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.933442.\n",
      "994 variants and 25309 people pass filters and QC.\n",
      "Among remaining phenotypes, 13034 are cases and 5191 are controls.  (7084\n",
      "phenotypes are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//MDE_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//MDE_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_MDE.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//MDE_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "203 variants loaded from .bim file.\n",
      "457 people (291 males, 166 females) loaded from .fam.\n",
      "426 phenotype values loaded from .fam.\n",
      "--keep: 457 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 457 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.985276.\n",
      "Warning: --hwe observation counts vary by more than 10%.  Consider using\n",
      "--geno, and/or applying different p-value thresholds to distinct subsets of\n",
      "your data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "90 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "113 variants and 457 people pass filters and QC.\n",
      "Among remaining phenotypes, 230 are cases and 196 are controls.  (31 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//MDE_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 4.97716.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (113 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//MDE_COMT.AllVariants.assoc.adjusted .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//MDE_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//MDE_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_MDE.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//MDE_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "203 variants loaded from .bim file.\n",
      "457 people (291 males, 166 females) loaded from .fam.\n",
      "426 phenotype values loaded from .fam.\n",
      "--keep: 457 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 457 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.985276.\n",
      "203 variants and 457 people pass filters and QC.\n",
      "Among remaining phenotypes, 230 are cases and 196 are controls.  (31 phenotypes\n",
      "are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//MDE_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//SAS_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//SAS_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_SAS.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//SAS_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "203 variants loaded from .bim file.\n",
      "539 people (351 males, 188 females) loaded from .fam.\n",
      "517 phenotype values loaded from .fam.\n",
      "--keep: 539 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 539 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.988366.\n",
      "Warning: --hwe observation counts vary by more than 10%.  Consider using\n",
      "--geno, and/or applying different p-value thresholds to distinct subsets of\n",
      "your data.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "83 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "120 variants and 539 people pass filters and QC.\n",
      "Among remaining phenotypes, 319 are cases and 198 are controls.  (22 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//SAS_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.10717.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (120 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//SAS_COMT.AllVariants.assoc.adjusted .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//SAS_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//SAS_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_SAS.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//SAS_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "203 variants loaded from .bim file.\n",
      "539 people (351 males, 188 females) loaded from .fam.\n",
      "517 phenotype values loaded from .fam.\n",
      "--keep: 539 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 539 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.988366.\n",
      "203 variants and 539 people pass filters and QC.\n",
      "Among remaining phenotypes, 319 are cases and 198 are controls.  (22 phenotypes\n",
      "are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//SAS_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//CAH_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --allow-no-sex\n",
      "  --assoc\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//CAH_COMT\n",
      "  --ci 0.95\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_CAH.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//CAH_COMT.AllVariants\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "366 variants loaded from .bim file.\n",
      "800 people (422 males, 378 females) loaded from .fam.\n",
      "777 phenotype values loaded from .fam.\n",
      "--keep: 800 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 800 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.994805.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "208 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "158 variants and 800 people pass filters and QC.\n",
      "Among remaining phenotypes, 494 are cases and 283 are controls.  (23 phenotypes\n",
      "are missing.)\n",
      "Writing C/C --assoc report to\n",
      "/home/jupyter/COMT_NBA_loop//CAH_COMT.AllVariants.assoc ...\n",
      "done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.47756.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (158 variants) written to\n",
      "/home/jupyter/COMT_NBA_loop//CAH_COMT.AllVariants.assoc.adjusted .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//CAH_COMT.AllVariants.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//CAH_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_CAH.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//CAH_COMT.AllVariants\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "366 variants loaded from .bim file.\n",
      "800 people (422 males, 378 females) loaded from .fam.\n",
      "777 phenotype values loaded from .fam.\n",
      "--keep: 800 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 800 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.994805.\n",
      "366 variants and 800 people pass filters and QC.\n",
      "Among remaining phenotypes, 494 are cases and 283 are controls.  (23 phenotypes\n",
      "are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//CAH_COMT.AllVariants.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis using plink assoc for ALL variants, not adjusting for any covariates\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "for ancestry in ancestries:\n",
    "\n",
    "    \n",
    "    ! /home/jupyter/plink1.9 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_COMT \\\n",
    "    --keep {WORK_DIR}/samplestokeep_{ancestry}.txt \\\n",
    "    --assoc \\\n",
    "    --maf 0.01 \\\n",
    "    --mac 2 \\\n",
    "    --hwe 0.0001 \\\n",
    "    --adjust \\\n",
    "    --allow-no-sex \\\n",
    "    --ci 0.95 \\\n",
    "    --out {WORK_DIR}/{ancestry}_COMT.AllVariants\n",
    "\n",
    "    #--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A).\n",
    "    ! /home/jupyter/plink1.9 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_COMT \\\n",
    "    --keep {WORK_DIR}/samplestokeep_{ancestry}.txt \\\n",
    "    --recode A \\\n",
    "    --out {WORK_DIR}/{ancestry}_COMT.AllVariants\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### glm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants_a0ref.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EUR_COMT\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/COMT_NBA_loop//EUR_covariate_file.txt\n",
      "  --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm a0-ref\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EUR.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants_a0ref\n",
      "\n",
      "Start time: Fri Mar 14 19:32:05 2025\n",
      "7445 MiB RAM detected, ~5184 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//EUR_COMT.fam.\n",
      "994 variants loaded from /home/jupyter/COMT_NBA_loop//EUR_COMT.bim.\n",
      "1 binary phenotype loaded (13034 cases, 5191 controls).\n",
      "--keep: 25309 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/COMT_NBA_loop//EUR_covariate_file.txt.\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) remaining after main\n",
      "filters.\n",
      "13034 cases and 5191 controls remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "Warning: --hwe filter is suspiciously strict for the sample size; you may be\n",
      "filtering out many variants with association signals.\n",
      "If you are performing routine QC, consider using --hwe with a sample-size term\n",
      "(e.g. \"--hwe 1e-5 0.001\" results in a threshold of 1e-6 with 1000 observations,\n",
      "1e-7 with 2000 observations, etc.), and/or specifying 'keep-fewhet' to remove\n",
      "only variants with heterozygosity excess.\n",
      "Alternatively, if the strictness is intentional (e.g. you are preparing data\n",
      "for a method that requires variants to be near Hardy-Weinberg equilibrium),\n",
      "explicitly specify a sample-size term of 0.\n",
      "This warning will be upgraded to an error in a future build.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "877 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "117 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': done.\n",
      "Results written to /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants_a0ref.PHENO1.glm.logistic.hybrid .\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.89459.\n",
      "--adjust values (117 tests) written to\n",
      "/home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants_a0ref.PHENO1.glm.logistic.hybrid.adjusted\n",
      ".\n",
      "End time: Fri Mar 14 19:32:05 2025\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants_a0ref.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EUR_COMT\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EUR.txt\n",
      "  --out /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants_a0ref\n",
      "  --recode A\n",
      "\n",
      "7445 MB RAM detected; reserving 3722 MB for main workspace.\n",
      "994 variants loaded from .bim file.\n",
      "25309 people (15423 males, 9886 females) loaded from .fam.\n",
      "18225 phenotype values loaded from .fam.\n",
      "--keep: 25309 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 25309 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.933442.\n",
      "994 variants and 25309 people pass filters and QC.\n",
      "Among remaining phenotypes, 13034 are cases and 5191 are controls.  (7084\n",
      "phenotypes are missing.)\n",
      "--recode A to /home/jupyter/COMT_NBA_loop//EUR_COMT.AllVariants_a0ref.raw ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#Run case-control analysis for all variants except introns, with covariates\n",
    "#ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "# Add a0-ref flag to glm argument on selected ancestries to make sure all the ancestries test the same allele\n",
    "ancestries_a0ref = ['EUR','AJ','MDE','SAS','CAH','CAS']\n",
    "\n",
    "for ancestry in ancestries_a0ref:\n",
    "    \n",
    "    ! /home/jupyter/plink2 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_COMT \\\n",
    "    --keep {WORK_DIR}/samplestokeep_{ancestry}.txt \\\n",
    "    --glm a0-ref \\\n",
    "    --ci 0.95 \\\n",
    "    --mac 2 \\\n",
    "    --maf 0.01 \\\n",
    "    --hwe 0.0001 \\\n",
    "    --adjust \\\n",
    "    --covar {WORK_DIR}/{ancestry}_covariate_file.txt \\\n",
    "    --covar-name SEX,AGE,PC1,PC2,PC3,PC4,PC5 \\\n",
    "    --covar-variance-standardize \\\n",
    "    --out {WORK_DIR}/{ancestry}_COMT.AllVariants_a0ref # --out {WORK_DIR}/{ancestry}_COMT.AllVariants in case no a0-ref flag is needed\n",
    "    \n",
    "    #--recode A creates a new text fileset, showing each variant in each case and control for the minor allele (A). \n",
    "    ! /home/jupyter/plink1.9 \\\n",
    "    --bfile {WORK_DIR}/{ancestry}_COMT \\\n",
    "    --keep {WORK_DIR}/samplestokeep_{ancestry}.txt \\\n",
    "    --recode A \\\n",
    "    --out {WORK_DIR}/{ancestry}_COMT.AllVariants_a0ref\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/EUR_COMT.AllVariants_a0ref.PHENO1.glm.logistic.hybrid {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "ancestry_GROUPS = ['AAC', 'AFR', 'AJ', 'CAS', 'EAS', 'EUR', 'SAS','CAH','MDE']\n",
    "\n",
    "for ancestry_GROUP in ancestry_GROUPS:        \n",
    "       shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}_COMT.AllVariants.PHENO1.glm.logistic.hybrid.adjusted {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/bonf/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "ancestry_GROUPS = ['AAC', 'AFR', 'AJ', 'CAS', 'EAS', 'EUR', 'SAS','CAH','MDE']\n",
    "\n",
    "for ancestry_GROUP in ancestry_GROUPS:        \n",
    "       shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}_COMT.AllVariants_a0ref.PHENO1.glm.logistic.hybrid.adjusted {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/bonf/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WORKING ON: AAC\n",
      "There are 9 variants with p-value < 0.05 in glm\n",
      "Number of variants in AAC for COMT: 413\n",
      "df_results shape: (413, 14)\n",
      "Merged dataframe shape: (186, 23)\n",
      "WORKING ON: AFR\n",
      "There are 1 variants with p-value < 0.05 in glm\n",
      "Number of variants in AFR for COMT: 487\n",
      "df_results shape: (487, 14)\n",
      "Merged dataframe shape: (181, 23)\n",
      "WORKING ON: AJ\n",
      "There are 5 variants with p-value < 0.05 in glm\n",
      "Number of variants in AJ for COMT: 245\n",
      "df_results shape: (245, 14)\n",
      "Merged dataframe shape: (128, 23)\n",
      "WORKING ON: AMR\n",
      "There are 13 variants with p-value < 0.05 in glm\n",
      "Number of variants in AMR for COMT: 249\n",
      "df_results shape: (249, 14)\n",
      "Merged dataframe shape: (124, 23)\n",
      "WORKING ON: CAS\n",
      "There are 12 variants with p-value < 0.05 in glm\n",
      "Number of variants in CAS for COMT: 229\n",
      "df_results shape: (229, 14)\n",
      "Merged dataframe shape: (121, 23)\n",
      "WORKING ON: EAS\n",
      "There are 3 variants with p-value < 0.05 in glm\n",
      "Number of variants in EAS for COMT: 420\n",
      "df_results shape: (420, 14)\n",
      "Merged dataframe shape: (115, 23)\n",
      "WORKING ON: EUR\n",
      "There are 9 variants with p-value < 0.05 in glm\n",
      "Number of variants in EUR for COMT: 994\n",
      "df_results shape: (994, 14)\n",
      "Merged dataframe shape: (117, 23)\n",
      "WORKING ON: MDE\n",
      "There are 7 variants with p-value < 0.05 in glm\n",
      "Number of variants in MDE for COMT: 203\n",
      "df_results shape: (203, 14)\n",
      "Merged dataframe shape: (111, 23)\n",
      "WORKING ON: SAS\n",
      "There are 6 variants with p-value < 0.05 in glm\n",
      "Number of variants in SAS for COMT: 203\n",
      "df_results shape: (203, 14)\n",
      "Merged dataframe shape: (120, 23)\n",
      "WORKING ON: CAH\n",
      "There are 6 variants with p-value < 0.05 in glm\n",
      "Number of variants in CAH for COMT: 366\n",
      "df_results shape: (366, 14)\n",
      "Merged dataframe shape: (158, 23)\n"
     ]
    }
   ],
   "source": [
    "#Process results from plink glm analysis for the variants\n",
    "#ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS', 'CAH']\n",
    "\n",
    "# We process the results in two batches depending on if we have added a0-ref flag or not.\n",
    "ancestries_a0ref = ['EUR','AJ','MDE','SAS','CAH','CAS']\n",
    "\n",
    "\n",
    "for ancestry in ancestries_a0ref: # for ancestry in ancestries:\n",
    "    \n",
    "    print(f'WORKING ON: {ancestry}')\n",
    "    \n",
    "    #Read in glm results\n",
    "    assoc = pd.read_csv(f'{WORK_DIR}/{ancestry}_COMT.AllVariants.PHENO1.glm.logistic.hybrid', delim_whitespace=True) # Change allExceptIntronic for allExceptIntronic_a0ref\n",
    "    assoc_add = assoc[assoc['TEST']==\"ADD\"]\n",
    "    \n",
    "    #Filter for significant variants p < 0.05 - if any\n",
    "    significant = assoc_add[assoc_add['P']<0.05]\n",
    "    print(f'There are {len(significant)} variants with p-value < 0.05 in glm')\n",
    "    \n",
    "    #Read in plink recoded data (.raw file)\n",
    "    recode = pd.read_csv(f'{WORK_DIR}/{ancestry}_COMT.AllVariants.raw', delim_whitespace=True)\n",
    "\n",
    "    # Make a list from the column names\n",
    "    column_names = recode.columns.tolist()\n",
    "\n",
    "    # Drop the first 6 columns to keep the variants \n",
    "    variants = column_names[6:]\n",
    "\n",
    "    print(f'Number of variants in {ancestry} for COMT: {len(variants)}')\n",
    "\n",
    "    # Pre-filter the dataset\n",
    "    cases_data = recode[recode['PHENOTYPE'] == 2]\n",
    "    controls_data = recode[recode['PHENOTYPE'] == 1]\n",
    "\n",
    "    results = []\n",
    "\n",
    "    # Pre-filter the dataset\n",
    "    total_cases = cases_data.shape[0]\n",
    "    total_controls = controls_data.shape[0]\n",
    "    results = []\n",
    "\n",
    "    for variant in variants:\n",
    "        ## For PD cases\n",
    "        hom_cases = (cases_data[variant] == 2).sum()\n",
    "        het_cases = (cases_data[variant] == 1).sum()\n",
    "        hom_ref_cases = (cases_data[variant] == 0).sum()  # Homozygous reference genotype\n",
    "        missing_cases = total_cases - (hom_cases + het_cases + hom_ref_cases)  # Missing data count\n",
    "        freq_cases = (2 * hom_cases + het_cases) / (2 * (total_cases - missing_cases))  # Adjust for missing data in denominator\n",
    "\n",
    "        ## For controls\n",
    "        hom_controls = (controls_data[variant] == 2).sum()\n",
    "        het_controls = (controls_data[variant] == 1).sum()\n",
    "        hom_ref_controls = (controls_data[variant] == 0).sum()  # Homozygous reference genotype\n",
    "        missing_controls = total_controls - (hom_controls + het_controls + hom_ref_controls)  # Missing data count\n",
    "        freq_controls = (2 * hom_controls + het_controls) / (2 * (total_controls - missing_controls))  # Adjust for missing data in denominator\n",
    "    \n",
    "        # Append results in dictionary format\n",
    "        results.append({\n",
    "            'Variant': variant,\n",
    "            'Hom Cases': hom_cases,\n",
    "            'Het Cases': het_cases,\n",
    "            'Hom Ref Cases': hom_ref_cases,\n",
    "            'Missing Cases': missing_cases,\n",
    "            'Total Cases': total_cases,\n",
    "            'Carrier Freq in Cases': freq_cases,\n",
    "            'Hom Controls': hom_controls,\n",
    "            'Het Controls': het_controls,\n",
    "            'Hom Ref Controls': hom_ref_controls,\n",
    "            'Missing Controls': missing_controls,\n",
    "            'Total Controls': total_controls,\n",
    "            'Carrier Freq in Controls': freq_controls\n",
    "        })\n",
    "\n",
    "    # Return\n",
    "    df_results = pd.DataFrame(results)\n",
    "    df_results['ID'] = df_results['Variant'].apply(lambda x: x.rsplit('_', 1)[0])\n",
    "\n",
    "    #Print dimensions of the df_results dataframe\n",
    "    print(f'df_results shape: {df_results.shape}')\n",
    "    \n",
    "    #Merge with the glm file\n",
    "    sig_merge = assoc_add[['ID','A1','A1_FREQ','OBS_CT','OR','LOG(OR)_SE','L95','U95','Z_STAT','P']]\n",
    "    merged = pd.merge(df_results, sig_merge, on='ID', how='right')\n",
    "    \n",
    "    #Print dimensions of the merged dataframe (just adding more columns)\n",
    "    print(f'Merged dataframe shape: {merged.shape}')\n",
    "    ## Save to CSV\n",
    "    merged.to_csv(f'{WORK_DIR}/{ancestry}.AllVariants_glm.txt', sep = '\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "ancestry_GROUPS = ['AAC', 'AFR', 'AJ', 'CAS', 'EAS', 'EUR', 'SAS','CAH','MDE']\n",
    "ancestry_GROUPS = ['AMR']\n",
    "\n",
    "for ancestry_GROUP in ancestry_GROUPS:        \n",
    "        shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}.AllVariants_glm.txt {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "ancestries_a0ref = ['EUR','AJ','MDE','SAS','CAH','AMR','CAS']\n",
    "\n",
    "for ancestry_GROUP in ancestries_a0ref:        \n",
    "        shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/{ancestry_GROUP}.AllVariants_a0ref_glm.txt {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/a0ref/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Clinical Data Analyses"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# Read clinical data\n",
    "full_clinic = pd.read_csv(f'{WORK_DIR}/extended_clinical_data_release7.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## MOCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2490, 8)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Create MOCA Dataset\n",
    "moca=full_clinic[['GP2ID','visit_name','visit_month','last_diagnosis','moca_total_score','education_level','education_years','age_at_baseline']]\n",
    "moca = moca.dropna(subset = ['moca_total_score'])\n",
    "moca.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Clinical data (num rows, num columns): (44281, 26)\n"
     ]
    }
   ],
   "source": [
    "# Let's load the master key\n",
    "FULL_PATH = WORK_DIR + f'master_key_release7_final.csv'\n",
    "key = pd.read_csv(FULL_PATH, low_memory=False)\n",
    "print(f'Clinical data (num rows, num columns): {key.shape}')\n",
    "pd.set_option('display.max_columns', None)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['GP2ID', 'GP2sampleID', 'study', 'study_arm', 'study_type', 'diagnosis',\n",
       "       'baseline_GP2_phenotype_for_qc', 'baseline_GP2_phenotype',\n",
       "       'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset',\n",
       "       'age_at_diagnosis', 'age_at_death', 'age_at_last_follow_up',\n",
       "       'race_for_qc', 'family_history_for_qc', 'region_for_qc', 'manifest_id',\n",
       "       'Genotyping_site', 'sample_type', 'GDPR', 'monogenic', 'label',\n",
       "       'pruned', 'pruned_reason', 'related'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "key.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "key = key[['GP2ID','GP2sampleID','label']]\n",
    "key_EUR = key[key['label']=='EUR']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "label\n",
       "EUR    1336\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moca_ancestry = pd.merge(key_EUR,moca,on='GP2ID')\n",
    "moca_ancestry['label'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "visit_month\n",
       "0.0     1201\n",
       "12.0     112\n",
       "24.0      19\n",
       "36.0       4\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moca_ancestry['visit_month'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1201, 10)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moca_ancestry_BL = moca_ancestry[moca_ancestry['visit_month']==0]\n",
    "moca_ancestry_BL.shape\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "covar = pd.read_csv(f'{WORK_DIR}/EUR_covariate_file.txt',sep=\"\\t\")\n",
    "covar_mocaBL=moca_ancestry_BL.merge(covar,on=\"GP2ID\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO\n",
       "2.0    928\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_mocaBL=covar_mocaBL[covar_mocaBL['PHENO']==2]\n",
    "covar_mocaBL['PHENO'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "595"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_mocaBL['education_level'].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "covar_mocaBL['education_level'] = covar_mocaBL['education_level'].replace({'NaN': '-9'})\n",
    "covar_mocaBL['education_level'] = covar_mocaBL['education_level'].replace({'High School/GED': '1'})\n",
    "covar_mocaBL['education_level'] = covar_mocaBL['education_level'].replace({'<High School': '1'})\n",
    "covar_mocaBL['education_level'] = covar_mocaBL['education_level'].replace({'Some college without degree': '2'})\n",
    "covar_mocaBL['education_level'] = covar_mocaBL['education_level'].replace({\"Bachelor's degree\": '3'})\n",
    "covar_mocaBL['education_level'] = covar_mocaBL['education_level'].replace({'Professional or doctoral degree': '4'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_mocaBL_final = covar_mocaBL[['FID','IID','FATID','MATID','SEX','PHENO','visit_name','moca_total_score','age_at_baseline','education_level','PC1','PC2','PC3','PC4','PC5']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_mocaBL_final = covar_mocaBL_final.drop_duplicates(subset=['IID'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_mocaBL_final.to_csv(f'{WORK_DIR}/Covar_MOCA_BL_EUR.txt',sep=\",\",index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_MOCA.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EUR_COMT\n",
      "  --ci 0.95\n",
      "  --covar /home/jupyter/COMT_NBA_loop//Covar_MOCA_BL_EUR.txt\n",
      "  --covar-name SEX,age_at_baseline,PC1,PC2,PC3,PC4,PC5,education_level\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EUR.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_MOCA\n",
      "  --pheno /home/jupyter/COMT_NBA_loop//Covar_MOCA_BL_EUR.txt\n",
      "  --pheno-name moca_total_score\n",
      "\n",
      "Start time: Fri Mar 28 16:53:52 2025\n",
      "7445 MiB RAM detected, ~5793 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//EUR_COMT.fam.\n",
      "994 variants loaded from /home/jupyter/COMT_NBA_loop//EUR_COMT.bim.\n",
      "1 quantitative phenotype loaded (927 values).\n",
      "--keep: 25309 samples remaining.\n",
      "8 covariates loaded from /home/jupyter/COMT_NBA_loop//Covar_MOCA_BL_EUR.txt.\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) remaining after main\n",
      "filters.\n",
      "927 quantitative phenotype values remaining after main filters.\n",
      "--covar-variance-standardize: 8 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "Warning: --hwe filter is suspiciously strict for the sample size; you may be\n",
      "filtering out many variants with association signals.\n",
      "If you are performing routine QC, consider using --hwe with a sample-size term\n",
      "(e.g. \"--hwe 1e-5 0.001\" results in a threshold of 1e-6 with 1000 observations,\n",
      "1e-7 with 2000 observations, etc.), and/or specifying 'keep-fewhet' to remove\n",
      "only variants with heterozygosity excess.\n",
      "Alternatively, if the strictness is intentional (e.g. you are preparing data\n",
      "for a method that requires variants to be near Hardy-Weinberg equilibrium),\n",
      "explicitly specify a sample-size term of 0.\n",
      "This warning will be upgraded to an error in a future build.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "877 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "117 variants remaining after main filters.\n",
      "--glm linear regression on phenotype 'moca_total_score': done.\n",
      "Results written to /home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_MOCA.moca_total_score.glm.linear .\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 2.64248.\n",
      "--adjust values (117 tests) written to\n",
      "/home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_MOCA.moca_total_score.glm.linear.adjusted\n",
      ".\n",
      "End time: Fri Mar 28 16:53:52 2025\n"
     ]
    }
   ],
   "source": [
    "!/home/jupyter/plink2 \\\n",
    "--bfile {WORK_DIR}/EUR_COMT \\\n",
    "--keep {WORK_DIR}/samplestokeep_EUR.txt \\\n",
    "--pheno {WORK_DIR}/Covar_MOCA_BL_EUR.txt \\\n",
    "--pheno-name moca_total_score \\\n",
    "--ci 0.95 \\\n",
    "--mac 2 \\\n",
    "--hwe 0.0001 \\\n",
    "--maf 0.01 \\\n",
    "--glm \\\n",
    "--adjust \\\n",
    "--covar {WORK_DIR}/Covar_MOCA_BL_EUR.txt \\\n",
    "--covar-name SEX,age_at_baseline,PC1,PC2,PC3,PC4,PC5,education_level \\\n",
    "--covar-variance-standardize \\\n",
    "--out {WORK_DIR}/COMT_GP2_EUR_MOCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "Moca_glm_add_AllVariants = pd.read_csv(f'{WORK_DIR}/COMT_GP2_EUR_MOCA.moca_total_score.glm.linear', delim_whitespace=True)\n",
    "Moca_glm_add_AllVariants_add=Moca_glm_add_AllVariants[(Moca_glm_add_AllVariants['TEST']=='ADD')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "Moca_glm_add_AllVariants_adj = pd.read_csv(f'{WORK_DIR}/COMT_GP2_EUR_MOCA.moca_total_score.glm.linear.adjusted', delim_whitespace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/COMT_GP2_EUR_MOCA.moca_total_score.glm.linear {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/MOCA/COMT_GP2_EUR_MOCA.moca_total_score.glm.linear')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/COMT_GP2_EUR_MOCA.moca_total_score.glm.linear.adjusted {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/MOCA/COMT_GP2_EUR_MOCA.moca_total_score.glm.linear.adjusted')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## UPDRS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "updrs = full_clinic[['GP2ID','visit_name','visit_month','mds_updrs_part_i_summary_score','mds_updrs_part_ii_summary_score','mds_updrs_part_iii_summary_score','mds_updrs_part_iv_summary_score','age_at_baseline']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "visit_month\n",
       " 0.0     14684\n",
       " 12.0     2443\n",
       " 24.0     2379\n",
       " 36.0     2066\n",
       " 6.0      1480\n",
       "         ...  \n",
       " 43.0        2\n",
       " 16.0        1\n",
       " 34.0        1\n",
       " 81.0        1\n",
       "-21.0        1\n",
       "Name: count, Length: 89, dtype: int64"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "updrs['visit_month'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "label\n",
       "EUR    20619\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "updrs_ancestry = pd.merge(key_EUR,updrs,on='GP2ID')\n",
    "updrs_ancestry['label'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "visit_month\n",
       " 0.0     8449\n",
       " 12.0    1647\n",
       " 24.0    1620\n",
       " 36.0    1495\n",
       " 6.0      894\n",
       "         ... \n",
       " 81.0       1\n",
       " 32.0       1\n",
       " 45.0       1\n",
       " 34.0       1\n",
       "-21.0       1\n",
       "Name: count, Length: 89, dtype: int64"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "updrs_ancestry['visit_month'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(8449, 10)"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "updrs_ancestry_BL = updrs_ancestry[updrs_ancestry['visit_month']==0]\n",
    "updrs_ancestry_BL.shape\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar = pd.read_csv(f'{WORK_DIR}/EUR_covariate_file.txt',sep=\"\\t\")\n",
    "covar_updrsBL=updrs_ancestry_BL.merge(covar,on=\"GP2ID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "PHENO\n",
       "2.0    4939\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_updrsBL_PD=covar_updrsBL[covar_updrsBL['PHENO']==2]\n",
    "covar_updrsBL_PD['PHENO'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_updrsBL_final = covar_updrsBL_PD[['FID','IID','FATID','MATID','SEX','PHENO','visit_name','mds_updrs_part_i_summary_score','mds_updrs_part_ii_summary_score','mds_updrs_part_iii_summary_score','mds_updrs_part_iv_summary_score','age_at_baseline','PC1','PC2','PC3','PC4','PC5']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_updrsBL_final = covar_updrsBL_final.drop_duplicates(subset=['IID'])\n",
    "covar_updrsBL_final.to_csv(f'{WORK_DIR}/Covar_UPDRS_BL_EUR.txt',sep=\",\",index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Part III"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_updrsBL_final=pd.read_csv(f'{WORK_DIR}/Covar_UPDRS_BL_EUR.txt',sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1268, 17)"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_updrsIII_BL_final = covar_updrsBL_final.dropna(subset = ['mds_updrs_part_iii_summary_score'])\n",
    "covar_updrsIII_BL_final.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_updrsIII_BL_final.to_csv(f'{WORK_DIR}/Covar_UPDRS_III_BL_EUR.txt',sep=\",\",index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FID                                 1268\n",
       "IID                                 1268\n",
       "FATID                               1268\n",
       "MATID                               1268\n",
       "SEX                                 1268\n",
       "PHENO                               1268\n",
       "visit_name                          1266\n",
       "mds_updrs_part_i_summary_score       670\n",
       "mds_updrs_part_ii_summary_score      668\n",
       "mds_updrs_part_iii_summary_score    1268\n",
       "mds_updrs_part_iv_summary_score      359\n",
       "age_at_baseline                     1261\n",
       "PC1                                 1268\n",
       "PC2                                 1268\n",
       "PC3                                 1268\n",
       "PC4                                 1268\n",
       "PC5                                 1268\n",
       "dtype: int64"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_updrsIII_BL_final.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_UPDRSIII_AllVariants.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EUR_COMT\n",
      "  --covar /home/jupyter/COMT_NBA_loop//Covar_UPDRS_III_BL_EUR.txt\n",
      "  --covar-name SEX,age_at_baseline,PC1,PC2,PC3,PC4,PC5\n",
      "  --covar-variance-standardize\n",
      "  --glm\n",
      "  --hwe 0.0001\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EUR.txt\n",
      "  --mac 2\n",
      "  --maf 0.01\n",
      "  --out /home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_UPDRSIII_AllVariants\n",
      "  --pheno /home/jupyter/COMT_NBA_loop//Covar_UPDRS_III_BL_EUR.txt\n",
      "  --pheno-name mds_updrs_part_iii_summary_score\n",
      "\n",
      "Start time: Fri Mar 28 17:10:44 2025\n",
      "7445 MiB RAM detected, ~5821 available; reserving 3722 MiB for main workspace.\n",
      "Using up to 2 compute threads.\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) loaded from\n",
      "/home/jupyter/COMT_NBA_loop//EUR_COMT.fam.\n",
      "994 variants loaded from /home/jupyter/COMT_NBA_loop//EUR_COMT.bim.\n",
      "1 quantitative phenotype loaded (1268 values).\n",
      "--keep: 25309 samples remaining.\n",
      "7 covariates loaded from /home/jupyter/COMT_NBA_loop//Covar_UPDRS_III_BL_EUR.txt.\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) remaining after main\n",
      "filters.\n",
      "1268 quantitative phenotype values remaining after main filters.\n",
      "--covar-variance-standardize: 7 covariates transformed.\n",
      "Calculating allele frequencies... done.\n",
      "Warning: --hwe filter is suspiciously strict for the sample size; you may be\n",
      "filtering out many variants with association signals.\n",
      "If you are performing routine QC, consider using --hwe with a sample-size term\n",
      "(e.g. \"--hwe 1e-5 0.001\" results in a threshold of 1e-6 with 1000 observations,\n",
      "1e-7 with 2000 observations, etc.), and/or specifying 'keep-fewhet' to remove\n",
      "only variants with heterozygosity excess.\n",
      "Alternatively, if the strictness is intentional (e.g. you are preparing data\n",
      "for a method that requires variants to be near Hardy-Weinberg equilibrium),\n",
      "explicitly specify a sample-size term of 0.\n",
      "This warning will be upgraded to an error in a future build.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "877 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "117 variants remaining after main filters.\n",
      "--glm linear regression on phenotype 'mds_updrs_part_iii_summary_score': done.\n",
      "Results written to /home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_UPDRSIII_AllVariants.mds_updrs_part_iii_summary_score.glm.linear .\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 0.949782.\n",
      "(Treating lambda as 1 in GC-corrected p-value calculation.)\n",
      "--adjust values (117 tests) written to\n",
      "/home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_UPDRSIII_AllVariants.mds_updrs_part_iii_summary_score.glm.linear.adjusted\n",
      ".\n",
      "End time: Fri Mar 28 17:10:44 2025\n"
     ]
    }
   ],
   "source": [
    "!/home/jupyter/plink2 \\\n",
    "--bfile {WORK_DIR}/EUR_COMT \\\n",
    "--keep {WORK_DIR}/samplestokeep_EUR.txt \\\n",
    "--pheno {WORK_DIR}/Covar_UPDRS_III_BL_EUR.txt \\\n",
    "--pheno-name mds_updrs_part_iii_summary_score \\\n",
    "--glm \\\n",
    "--adjust \\\n",
    "--mac 2 \\\n",
    "--maf 0.01 \\\n",
    "--hwe 0.0001 \\\n",
    "--covar {WORK_DIR}/Covar_UPDRS_III_BL_EUR.txt \\\n",
    "--covar-name SEX,age_at_baseline,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--out {WORK_DIR}/COMT_GP2_EUR_UPDRSIII_AllVariants\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "UPDRSiii_glm_add_AllVariants = pd.read_csv(f'{WORK_DIR}/COMT_GP2_EUR_UPDRSIII_AllVariants.mds_updrs_part_iii_summary_score.glm.linear', delim_whitespace=True)\n",
    "UPDRSiii_glm_add_AllVariants_add=UPDRSiii_glm_add_AllVariants[(UPDRSiii_glm_add_AllVariants['TEST']=='ADD')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "UPDRSiii_glm_add_AllVariants_adjusted = pd.read_csv(f'{WORK_DIR}/COMT_GP2_EUR_UPDRSIII_all_variants_except_intronic.mds_updrs_part_iii_summary_score.glm.linear.adjusted', delim_whitespace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/COMT_GP2_EUR_UPDRSIII_AllVariants.mds_updrs_part_iii_summary_score.glm.linear {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/UPDRS/COMT_GP2_EUR_UPDRSIII_AllVariants.mds_updrs_part_iii_summary_score.glm.linear')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/COMT_GP2_EUR_UPDRSIII_AllVariants.mds_updrs_part_iii_summary_score.glm.linear.adjusted {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/UPDRS/COMT_GP2_EUR_UPDRSIII_AllVariants.mds_updrs_part_iii_summary_score.glm.linear.adjusted')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Part IV"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_updrsBL_final=pd.read_csv(f'{WORK_DIR}/Covar_UPDRS_BL_EUR.txt',sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(359, 17)"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "covar_updrsIV_BL_final = covar_updrsBL_final.dropna(subset = ['mds_updrs_part_iv_summary_score'])\n",
    "covar_updrsIV_BL_final.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_updrsIV_BL_final.to_csv(f'{WORK_DIR}/Covar_UPDRS_IV_BL_EUR.txt',sep=\",\",index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "covar_updrsIV_BL_final.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\r\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\r\n",
      "Logging to /home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_UPDRSIV_AllVariants.log.\r\n",
      "Options in effect:\r\n",
      "  --adjust\r\n",
      "  --bfile /home/jupyter/COMT_NBA_loop//EUR_COMT\r\n",
      "  --covar /home/jupyter/COMT_NBA_loop//Covar_UPDRS_IV_BL_EUR.txt\r\n",
      "  --covar-name SEX,age_at_baseline,PC1,PC2,PC3,PC4,PC5\r\n",
      "  --covar-variance-standardize\r\n",
      "  --glm\r\n",
      "  --hwe 0.0001\r\n",
      "  --keep /home/jupyter/COMT_NBA_loop//samplestokeep_EUR.txt\r\n",
      "  --mac 2\r\n",
      "  --maf 0.01\r\n",
      "  --out /home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_UPDRSIV_AllVariants\r\n",
      "  --pheno /home/jupyter/COMT_NBA_loop//Covar_UPDRS_IV_BL_EUR.txt\r\n",
      "  --pheno-name mds_updrs_part_iv_summary_score\r\n",
      "\r\n",
      "Start time: Fri Mar 28 17:13:05 2025\r\n",
      "7445 MiB RAM detected, ~5793 available; reserving 3722 MiB for main workspace.\r\n",
      "Using up to 2 compute threads.\r\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) loaded from\r\n",
      "/home/jupyter/COMT_NBA_loop//EUR_COMT.fam.\r\n",
      "994 variants loaded from /home/jupyter/COMT_NBA_loop//EUR_COMT.bim.\r\n",
      "1 quantitative phenotype loaded (359 values).\r\n",
      "--keep: 25309 samples remaining.\r\n",
      "7 covariates loaded from /home/jupyter/COMT_NBA_loop//Covar_UPDRS_IV_BL_EUR.txt.\r\n",
      "25309 samples (9886 females, 15423 males; 25309 founders) remaining after main\r\n",
      "filters.\r\n",
      "359 quantitative phenotype values remaining after main filters.\r\n",
      "--covar-variance-standardize: 7 covariates transformed.\r\n",
      "Calculating allele frequencies... 0%\b\bdone.\r\n",
      "Warning: --hwe filter is suspiciously strict for the sample size; you may be\r\n",
      "filtering out many variants with association signals.\r\n",
      "If you are performing routine QC, consider using --hwe with a sample-size term\r\n",
      "(e.g. \"--hwe 1e-5 0.001\" results in a threshold of 1e-6 with 1000 observations,\r\n",
      "1e-7 with 2000 observations, etc.), and/or specifying 'keep-fewhet' to remove\r\n",
      "only variants with heterozygosity excess.\r\n",
      "Alternatively, if the strictness is intentional (e.g. you are preparing data\r\n",
      "for a method that requires variants to be near Hardy-Weinberg equilibrium),\r\n",
      "explicitly specify a sample-size term of 0.\r\n",
      "This warning will be upgraded to an error in a future build.\r\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\r\n",
      "877 variants removed due to allele frequency threshold(s)\r\n",
      "(--maf/--max-maf/--mac/--max-mac).\r\n",
      "117 variants remaining after main filters.\r\n",
      "--glm linear regression on phenotype 'mds_updrs_part_iv_summary_score': 0%\b\bdone.\r\n",
      "Results written to /home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_UPDRSIV_AllVariants.mds_updrs_part_iv_summary_score.glm.linear .\r\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.90947.\r\n",
      "--adjust values (117 tests) written to\r\n",
      "/home/jupyter/COMT_NBA_loop//COMT_GP2_EUR_UPDRSIV_AllVariants.mds_updrs_part_iv_summary_score.glm.linear.adjusted\r\n",
      ".\r\n",
      "End time: Fri Mar 28 17:13:05 2025\r\n"
     ]
    }
   ],
   "source": [
    "!/home/jupyter/plink2 \\\n",
    "--bfile {WORK_DIR}/EUR_COMT \\\n",
    "--keep {WORK_DIR}/samplestokeep_EUR.txt \\\n",
    "--pheno {WORK_DIR}/Covar_UPDRS_IV_BL_EUR.txt \\\n",
    "--pheno-name mds_updrs_part_iv_summary_score \\\n",
    "--glm \\\n",
    "--adjust \\\n",
    "--mac 2 \\\n",
    "--hwe 0.0001 \\\n",
    "--maf 0.01 \\\n",
    "--covar {WORK_DIR}/Covar_UPDRS_IV_BL_EUR.txt \\\n",
    "--covar-name SEX,age_at_baseline,PC1,PC2,PC3,PC4,PC5 \\\n",
    "--covar-variance-standardize \\\n",
    "--out {WORK_DIR}/COMT_GP2_EUR_UPDRSIV_AllVariants\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "UPDRSiv_glm_add_AllVariants = pd.read_csv(f'{WORK_DIR}/COMT_GP2_EUR_UPDRSIV_all_variants_except_intronic.mds_updrs_part_iv_summary_score.glm.linear', delim_whitespace=True)\n",
    "UPDRSiv_glm_add_AllVariants_add=UPDRSiv_glm_add_AllVariants[(UPDRSiv_glm_add_AllVariants['TEST']=='ADD')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "Covar_UPDRS_IV_BL_EURUPDRSiv_glm_add_AllVariants_add_adjusted = pd.read_csv(f'{WORK_DIR}/COMT_GP2_EUR_UPDRSIV_all_variants_except_intronic.mds_updrs_part_iv_summary_score.glm.linear.adjusted', delim_whitespace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/COMT_GP2_EUR_UPDRSIV_AllVariants.mds_updrs_part_iv_summary_score.glm.linear {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/UPDRS/COMT_GP2_EUR_UPDRSIV_AllVariants.mds_updrs_part_iv_summary_score.glm.linear')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/COMT_GP2_EUR_UPDRSIV_AllVariants.mds_updrs_part_iv_summary_score.glm.linear.adjusted {WORKSPACE_BUCKET}/COMT_NBA_loop/GLM_tables/UPDRS/COMT_GP2_EUR_UPDRSIV_AllVariants.mds_updrs_part_iv_summary_score.glm.linear.adjusted')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## LIDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "In addition: Warning message:\n",
       "In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
       "  library ‘/usr/lib/R/site-library’ contains no packages\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "BILLING_PROJECT_ID <- Sys.getenv('GOOGLE_PROJECT')\n",
    "WORKSPACE_NAMESPACE <- Sys.getenv('WORKSPACE_NAMESPACE')\n",
    "WORKSPACE_NAME <- Sys.getenv('WORKSPACE_NAME')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "data.table 1.16.2 using 1 threads (see ?getDTthreads).  Latest news: r-datatable.com\n",
       "\n",
       "Attaching package: ‘dplyr’\n",
       "\n",
       "The following objects are masked from ‘package:data.table’:\n",
       "\n",
       "    between, first, last\n",
       "\n",
       "The following objects are masked from ‘package:stats’:\n",
       "\n",
       "    filter, lag\n",
       "\n",
       "The following objects are masked from ‘package:base’:\n",
       "\n",
       "    intersect, setdiff, setequal, union\n",
       "\n",
       "Loading required package: ggpubr\n",
       "\n",
       "Attaching package: ‘survminer’\n",
       "\n",
       "The following object is masked from ‘package:survival’:\n",
       "\n",
       "    myeloma\n",
       "\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "library(data.table)\n",
    "library(dplyr)\n",
    "library(ggplot2)\n",
    "library(tidyr)\n",
    "library(survival)\n",
    "library(survminer)\n",
    "setwd(\"/home/jupyter/COMT_NBA_loop/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "master_key<-fread(\"master_key_release7_final.csv\") %>% \n",
    "              select(GP2ID,GP2sampleID,label)  %>%\n",
    "              filter(label==\"EUR\") %>% \n",
    "              na.exclude(dyskinesia_status_at_censoring)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 32548\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "full_clinic <- fread('extended_clinical_data_release7.csv')\n",
    "nrow(full_clinic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "lids <- full_clinic[,c('GP2ID','visit_name','visit_month','age_at_baseline','dyskinesia_status_at_censoring','age_at_dyskinesia_status_censored','ledd_daily')]\n",
    "lids[,'Disease_duration_fromBL'] = lids[,'age_at_dyskinesia_status_censored'] - lids[,'age_at_baseline']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "# Filter on EUR and also must have dyskinesia_status_at_censoring filled\n",
    "raw_eur <- left_join(lids,master_key,\"GP2ID\") %>%\n",
    "                        filter(label==\"EUR\") %>% \n",
    "                        na.exclude(dyskinesia_status_at_censoring)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "# Adding new columns: \n",
    "#   - visit_month_year: visit month on year format\n",
    "#   - age_at_visit: sum visit_month_year to age at baseline \n",
    "raw_eur_formatted<-raw_eur %>% mutate(visit_month_year=(visit_month)/12,\n",
    "                                      age_at_visit=(age_at_baseline+visit_month_year))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "# I will add LEDD as covariate at the moment of dyskinesia_status_at_censoring. \n",
    "# For this we have to compare age_at_dyskinesia_status_censored and age_at_visit. \n",
    "# We have to keep the closer visit to age_at_dyskinesia_status_censored.\n",
    "LIDs_SelectedVisit <-  raw_eur_formatted %>% \n",
    "                                      group_by(GP2ID) %>% \n",
    "                                      slice_min(abs(age_at_dyskinesia_status_censored - age_at_visit))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  0   1 \n",
      "295 154 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "table(LIDs_SelectedVisit$dyskinesia_status_at_censoring)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "#merge covar with subjects with LIDs data on EUR\n",
    "covar = fread('EUR_covariate_file.txt')\n",
    "covar_lids=left_join(LIDs_SelectedVisit,covar,\"GP2ID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  2 \n",
      "437 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "table(covar_lids$PHENO) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "#merge with Genetic\n",
    "COMT_GP2_all_variants_recode <- fread(\"EUR_COMT.AllVariants.raw\")\n",
    "covar_lids_comt <- left_join(covar_lids,COMT_GP2_all_variants_recode,'IID')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "lids <- covar_lids_comt\n",
    "\n",
    "latest_columns <- lids[, -(1:29)]\n",
    "rounded_columns <- round(latest_columns, 0)\n",
    "lids[, (30:ncol(lids))] <- rounded_columns\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "**rs4680**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call:\n",
      "coxph(formula = surv_object_LIDs ~ lids[[snp]] + age_at_dyskinesia_status_censored + \n",
      "    ledd_daily + SEX.x + PC1 + PC2 + PC3 + PC4 + PC5, data = lids)\n",
      "\n",
      "  n= 423, number of events= 147 \n",
      "   (26 observations deleted due to missingness)\n",
      "\n",
      "                                        coef  exp(coef)   se(coef)      z\n",
      "lids[[snp]]                       -5.361e-02  9.478e-01  1.257e-01 -0.426\n",
      "age_at_dyskinesia_status_censored -2.775e-02  9.726e-01  9.111e-03 -3.046\n",
      "ledd_daily                         3.637e-04  1.000e+00  1.831e-04  1.986\n",
      "SEX.x                              2.596e-01  1.296e+00  1.736e-01  1.495\n",
      "PC1                                1.431e+01  1.637e+06  1.727e+01  0.828\n",
      "PC2                                1.247e+01  2.611e+05  2.510e+01  0.497\n",
      "PC3                                2.276e+01  7.684e+09  1.549e+01  1.469\n",
      "PC4                                4.354e+00  7.776e+01  2.609e+01  0.167\n",
      "PC5                                7.169e+00  1.299e+03  1.422e+01  0.504\n",
      "                                  Pr(>|z|)   \n",
      "lids[[snp]]                        0.66985   \n",
      "age_at_dyskinesia_status_censored  0.00232 **\n",
      "ledd_daily                         0.04698 * \n",
      "SEX.x                              0.13485   \n",
      "PC1                                0.40743   \n",
      "PC2                                0.61920   \n",
      "PC3                                0.14177   \n",
      "PC4                                0.86748   \n",
      "PC5                                0.61416   \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "                                  exp(coef) exp(-coef) lower .95 upper .95\n",
      "lids[[snp]]                       9.478e-01  1.055e+00 7.408e-01 1.213e+00\n",
      "age_at_dyskinesia_status_censored 9.726e-01  1.028e+00 9.554e-01 9.902e-01\n",
      "ledd_daily                        1.000e+00  9.996e-01 1.000e+00 1.001e+00\n",
      "SEX.x                             1.296e+00  7.714e-01 9.225e-01 1.822e+00\n",
      "PC1                               1.637e+06  6.107e-07 3.251e-09 8.249e+20\n",
      "PC2                               2.611e+05  3.830e-06 1.134e-16 6.014e+26\n",
      "PC3                               7.684e+09  1.301e-10 4.990e-04 1.183e+23\n",
      "PC4                               7.776e+01  1.286e-02 4.801e-21 1.259e+24\n",
      "PC5                               1.299e+03  7.701e-04 1.021e-09 1.651e+15\n",
      "\n",
      "Concordance= 0.598  (se = 0.027 )\n",
      "Likelihood ratio test= 19.66  on 9 df,   p=0.02\n",
      "Wald test            = 19.47  on 9 df,   p=0.02\n",
      "Score (logrank) test = 19.6  on 9 df,   p=0.02\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "snp = 'chr22:19963748:G:A_G'\n",
    "\n",
    "surv_object_LIDs <- Surv(time = lids$Disease_duration_fromBL, event = lids$dyskinesia_status_at_censoring)\n",
    "\n",
    "model.cox_LIDs <- coxph(surv_object_LIDs ~ lids[[snp]] + age_at_dyskinesia_status_censored + ledd_daily + SEX.x + PC1 + PC2 + PC3 + PC4 + PC5, \n",
    "                   data = lids)\n",
    "summary(model.cox_LIDs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "**rs4818**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call:\n",
      "coxph(formula = surv_object_LIDs ~ lids[[snp]] + age_at_dyskinesia_status_censored + \n",
      "    SEX.x + ledd_daily + PC1 + PC2 + PC3 + PC4 + PC5, data = lids)\n",
      "\n",
      "  n= 429, number of events= 150 \n",
      "   (20 observations deleted due to missingness)\n",
      "\n",
      "                                        coef  exp(coef)   se(coef)      z\n",
      "lids[[snp]]                       -5.540e-02  9.461e-01  1.240e-01 -0.447\n",
      "age_at_dyskinesia_status_censored -2.699e-02  9.734e-01  9.042e-03 -2.984\n",
      "SEX.x                              2.540e-01  1.289e+00  1.733e-01  1.466\n",
      "ledd_daily                         3.854e-04  1.000e+00  1.800e-04  2.142\n",
      "PC1                                1.239e+01  2.415e+05  1.697e+01  0.731\n",
      "PC2                                1.553e+01  5.542e+06  2.492e+01  0.623\n",
      "PC3                                2.297e+01  9.449e+09  1.550e+01  1.482\n",
      "PC4                                5.297e+00  1.997e+02  2.555e+01  0.207\n",
      "PC5                                8.655e+00  5.738e+03  1.405e+01  0.616\n",
      "                                  Pr(>|z|)   \n",
      "lids[[snp]]                        0.65500   \n",
      "age_at_dyskinesia_status_censored  0.00284 **\n",
      "SEX.x                              0.14278   \n",
      "ledd_daily                         0.03221 * \n",
      "PC1                                0.46507   \n",
      "PC2                                0.53325   \n",
      "PC3                                0.13841   \n",
      "PC4                                0.83575   \n",
      "PC5                                0.53794   \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "                                  exp(coef) exp(-coef) lower .95 upper .95\n",
      "lids[[snp]]                       9.461e-01  1.057e+00 7.420e-01 1.206e+00\n",
      "age_at_dyskinesia_status_censored 9.734e-01  1.027e+00 9.563e-01 9.908e-01\n",
      "SEX.x                             1.289e+00  7.757e-01 9.179e-01 1.811e+00\n",
      "ledd_daily                        1.000e+00  9.996e-01 1.000e+00 1.001e+00\n",
      "PC1                               2.415e+05  4.141e-06 8.729e-10 6.680e+19\n",
      "PC2                               5.542e+06  1.804e-07 3.388e-15 9.066e+27\n",
      "PC3                               9.449e+09  1.058e-10 6.032e-04 1.480e+23\n",
      "PC4                               1.997e+02  5.009e-03 3.586e-20 1.112e+24\n",
      "PC5                               5.738e+03  1.743e-04 6.281e-09 5.242e+15\n",
      "\n",
      "Concordance= 0.597  (se = 0.027 )\n",
      "Likelihood ratio test= 19.76  on 9 df,   p=0.02\n",
      "Wald test            = 19.68  on 9 df,   p=0.02\n",
      "Score (logrank) test = 19.81  on 9 df,   p=0.02\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "snp = 'chr22:19963684:C:G_G'\n",
    "\n",
    "surv_object_LIDs <- Surv(time = lids$Disease_duration_fromBL, event = lids$dyskinesia_status_at_censoring)\n",
    "\n",
    "model.cox_LIDs <- coxph(surv_object_LIDs ~ lids[[snp]] + age_at_dyskinesia_status_censored + SEX.x + ledd_daily + PC1 + PC2 + PC3 + PC4 + PC5, \n",
    "                   data = lids)\n",
    "summary(model.cox_LIDs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "**rs4633**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call:\n",
      "coxph(formula = surv_object_LIDs ~ lids[[snp]] + age_at_dyskinesia_status_censored + \n",
      "    SEX.x + ledd_daily + PC1 + PC2 + PC3 + PC4 + PC5, data = lids)\n",
      "\n",
      "  n= 425, number of events= 148 \n",
      "   (24 observations deleted due to missingness)\n",
      "\n",
      "                                        coef  exp(coef)   se(coef)      z\n",
      "lids[[snp]]                       -6.837e-02  9.339e-01  1.252e-01 -0.546\n",
      "age_at_dyskinesia_status_censored -2.710e-02  9.733e-01  9.076e-03 -2.986\n",
      "SEX.x                              2.584e-01  1.295e+00  1.733e-01  1.490\n",
      "ledd_daily                         3.813e-04  1.000e+00  1.812e-04  2.104\n",
      "PC1                                1.251e+01  2.707e+05  1.700e+01  0.736\n",
      "PC2                                1.352e+01  7.406e+05  2.505e+01  0.540\n",
      "PC3                                2.188e+01  3.174e+09  1.550e+01  1.411\n",
      "PC4                                7.012e+00  1.110e+03  2.589e+01  0.271\n",
      "PC5                                7.825e+00  2.503e+03  1.419e+01  0.552\n",
      "                                  Pr(>|z|)   \n",
      "lids[[snp]]                        0.58501   \n",
      "age_at_dyskinesia_status_censored  0.00283 **\n",
      "SEX.x                              0.13611   \n",
      "ledd_daily                         0.03535 * \n",
      "PC1                                0.46189   \n",
      "PC2                                0.58949   \n",
      "PC3                                0.15818   \n",
      "PC4                                0.78651   \n",
      "PC5                                0.58125   \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "                                  exp(coef) exp(-coef) lower .95 upper .95\n",
      "lids[[snp]]                       9.339e-01  1.071e+00 7.307e-01 1.194e+00\n",
      "age_at_dyskinesia_status_censored 9.733e-01  1.027e+00 9.561e-01 9.907e-01\n",
      "SEX.x                             1.295e+00  7.723e-01 9.218e-01 1.819e+00\n",
      "ledd_daily                        1.000e+00  9.996e-01 1.000e+00 1.001e+00\n",
      "PC1                               2.707e+05  3.695e-06 9.138e-10 8.017e+19\n",
      "PC2                               7.406e+05  1.350e-06 3.537e-16 1.551e+27\n",
      "PC3                               3.174e+09  3.151e-10 2.020e-04 4.986e+22\n",
      "PC4                               1.110e+03  9.007e-04 1.018e-19 1.211e+25\n",
      "PC5                               2.503e+03  3.995e-04 2.099e-09 2.984e+15\n",
      "\n",
      "Concordance= 0.598  (se = 0.027 )\n",
      "Likelihood ratio test= 19.55  on 9 df,   p=0.02\n",
      "Wald test            = 19.51  on 9 df,   p=0.02\n",
      "Score (logrank) test = 19.64  on 9 df,   p=0.02\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "snp = 'chr22:19962712:C:T_C'\n",
    "\n",
    "surv_object_LIDs <- Surv(time = lids$Disease_duration_fromBL, event = lids$dyskinesia_status_at_censoring)\n",
    "\n",
    "model.cox_LIDs <- coxph(surv_object_LIDs ~ lids[[snp]] + age_at_dyskinesia_status_censored + SEX.x + ledd_daily + PC1 + PC2 + PC3 + PC4 + PC5, \n",
    "                   data = lids)\n",
    "summary(model.cox_LIDs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "**rs6269**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Call:\n",
      "coxph(formula = surv_object_LIDs ~ lids[[snp]] + age_at_dyskinesia_status_censored + \n",
      "    SEX.x + ledd_daily + PC1 + PC2 + PC3 + PC4 + PC5, data = lids)\n",
      "\n",
      "  n= 428, number of events= 150 \n",
      "   (21 observations deleted due to missingness)\n",
      "\n",
      "                                        coef  exp(coef)   se(coef)      z\n",
      "lids[[snp]]                       -2.772e-02  9.727e-01  1.234e-01 -0.225\n",
      "age_at_dyskinesia_status_censored -2.671e-02  9.736e-01  9.053e-03 -2.950\n",
      "SEX.x                              2.524e-01  1.287e+00  1.732e-01  1.457\n",
      "ledd_daily                         3.862e-04  1.000e+00  1.798e-04  2.149\n",
      "PC1                                1.216e+01  1.919e+05  1.693e+01  0.719\n",
      "PC2                                1.440e+01  1.786e+06  2.483e+01  0.580\n",
      "PC3                                2.230e+01  4.846e+09  1.558e+01  1.431\n",
      "PC4                                4.973e+00  1.444e+02  2.552e+01  0.195\n",
      "PC5                                8.992e+00  8.035e+03  1.404e+01  0.641\n",
      "                                  Pr(>|z|)   \n",
      "lids[[snp]]                        0.82220   \n",
      "age_at_dyskinesia_status_censored  0.00318 **\n",
      "SEX.x                              0.14505   \n",
      "ledd_daily                         0.03167 * \n",
      "PC1                                0.47234   \n",
      "PC2                                0.56204   \n",
      "PC3                                0.15230   \n",
      "PC4                                0.84549   \n",
      "PC5                                0.52176   \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "                                  exp(coef) exp(-coef) lower .95 upper .95\n",
      "lids[[snp]]                       9.727e-01  1.028e+00 7.638e-01 1.239e+00\n",
      "age_at_dyskinesia_status_censored 9.736e-01  1.027e+00 9.565e-01 9.911e-01\n",
      "SEX.x                             1.287e+00  7.769e-01 9.166e-01 1.807e+00\n",
      "ledd_daily                        1.000e+00  9.996e-01 1.000e+00 1.001e+00\n",
      "PC1                               1.919e+05  5.211e-06 7.504e-10 4.908e+19\n",
      "PC2                               1.786e+06  5.600e-07 1.315e-15 2.425e+27\n",
      "PC3                               4.846e+09  2.064e-10 2.655e-04 8.846e+22\n",
      "PC4                               1.444e+02  6.924e-03 2.752e-20 7.580e+23\n",
      "PC5                               8.035e+03  1.244e-04 9.079e-09 7.112e+15\n",
      "\n",
      "Concordance= 0.595  (se = 0.027 )\n",
      "Likelihood ratio test= 19.03  on 9 df,   p=0.02\n",
      "Wald test            = 18.94  on 9 df,   p=0.03\n",
      "Score (logrank) test = 19.05  on 9 df,   p=0.02\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "snp = 'chr22:19962429:A:G_G'\n",
    "\n",
    "surv_object_LIDs <- Surv(time = lids$Disease_duration_fromBL, event = lids$dyskinesia_status_at_censoring)\n",
    "\n",
    "model.cox_LIDs <- coxph(surv_object_LIDs ~ lids[[snp]] + age_at_dyskinesia_status_censored + SEX.x + ledd_daily + PC1 + PC2 + PC3 + PC4 + PC5, \n",
    "                   data = lids)\n",
    "summary(model.cox_LIDs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "**KAPLAN-MEIER CURVES**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAAC/VBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///9KoeSIAAAd0klEQVR4nO2deXwURb7AiyCiQVFcn9e64r596qrrKr3ueoK74oFGWB8uLAi6oBBgWWHBiIEAKvsMCKJymAaRw4TDRHIpERMiJAZydchJEkIGyYwMuSajwhoPdurzqma6c0xmMj3T3enu4vf9Y+ronkoNX7qnuqqmCmGAaZDeFQC0BQQzDghmHBDMOAoEVz4cPmQsxsuHDfr5rK870y3PXHHxxDbVKggoI3jB34vh8UunZOXyGN/xds6ua8Z1ph+644ui+59Qt5ZAyAQnmHtl1EVrd9008MoZGD8zujN/c7hLSjvQAXIxI4ualQRCJ0jBQ7JdlgHbvyrdjPHPYu6/bHixJ//t6zrSdlRErma0Xf2qAqEQpOBZGJed76DR02jQmsIplzTTuGPo8s70zWOcZ55BqzWoKxACQQpei/HZEUOe3vE9/gY9jfGPV71Lcv9935iznemK2/udN+3yt7WpLhAsQQp+j7z8J/ul67gfXeHLSPzeaIzbH3yINLw607itzYFSNagrEAIhCCa0ojL8wDPkar4mDn//yP3/pnlSmvJG+NfqVhMIleAFFy0v/2rNhW049fxNFbOGtLge/0VuaWnpWSmNP9yZ9/qAlRpVFwiW4AXXjvzZhcP2kgR/ffh9Am5HblqkNP7gugE3rNOmskDwQFcl44BgxgHBjAOCGQcEMw4IZhwQzDggmHGCEvxNG2Aevgte8BNLAdOw5KXgBc91AqahFQSzDQhmHBDMOCCYcUAw44BgxgHBjAOCGQcEMw4IZhwQzDggmHFAMOOAYMbxL/hs1Ph9nljaksV26RUEmwz/gl2OXR7BrVGuutfFVxBsNnq7RYuC87ZhHCm+YlySmDirZzEf/ZHwKGHN1q1bXyXhqAiCnRx5NcLDc/S0KWJiGYnbxHhEPEnkSYkvSCJeSthIYpkYn0Lf/7yYeIXET0lnbe2rfytTIkPw3iSMp4mvGL8zcuTzPYuJ50RW8DwfJSVOkiNzxfhoetrjYuJFErdKZ8WRRLaU2EcSvJRoIIkXxfjj9P1jxAS9h9ils9b11b+VKQn6CvZ9i67dScggFJWXlx/K+Hjx4k+zs7NbyZGSbA8H6WlfiInDJN4ixrNrSMIqJay0MCnRQhKHxTi9tJ0HxUQJiTs6318aMb6v/r1MRyDBLid2RLmOxoqvMr+Dm3j+K1p4U1NTM003k4iDhI6uGa1iRhPNcCj5EIXcA0rezjS9CF4+bdYGfGYGaT/HxNilV9mCK0lQSO7XO2h6K4mUkrCUhO6vzJ0kUkDCahJupBmpJxR8CBDsH22egx3b+MONjY0Hib7tJGykgotIWEQF04wdJJJHwnISbiDhLr6kLfQPAYL9o43gNqGgWBCEooKCgkISCoUkQjOKfWUUkDCez7SE/iFAsH806smqPhwcRHBt6B8CBPvHIF2ViXzmkZMhk8ONaFK/TmxgEMENZfRGHSq7ueECGPaNQQQ7GyvLQ2ffjH8IpzSoFAsYRbBCSkGwH0Aw4xhEcFFOUbOS98etrlarKoxhEMFJPL9PyftHcorezjAGEVy+nf/U6bQeOnSIdmA6y0jkGAntJDxEu6krSEgflBtpBh2DqCbhEVp/mtEIgv1iEMHOY1RXFc/z79FUEu3JJGEdCXk6KJFOws9J2EAz6DDzXhLuJeEpmnECBPvFKILdWPbs2fMpjeSSSAUJG0i4h17B+SSkY4wnaQZ95C0mYREJm2nGSeeD3KZCmzq0aPXp9MFQgkNnBPcKn6egq6QL5Xp/FnVhRPCDRHBBpQqUC6V6fxZ1YUTwI1yGpVWNgk6BYIMKzqW9HYK7S7qehHQeUBu949IOkOMkrKKnlZAInWrSIN2Ky0mEzvyyltg9BZ0ShCAHwtyUK3qK1xJGBB+tbnIWbCINamrrcxKmk7CVtrDrSCSPhEn0tPdIhJoukqaafEAidIJXOZ/qKailJLRv7gZdPrYMGBFMOZhCoNdnMQnJBe100Aw6FaiMhNn0lHQSqSdhFQk/oxkZJEIv9nI+USylpTEEqkCwwekUHBJHhJoGBVOOtAQEu6lO+FjR28XvdgPCiOA3llbp+efttaWCTc8K+IcRwe5WtJ5UgWBNUS64gVdUAgjWFuWCS/lkJW8HwdqiXHAJv03J20GwtjzCzeaVPadU84p+hgqCteUJjuNKlRVRX+d0ZGWFOpYEgrUlZ+VKq/JSWnn+/RDfCoLNgCPTPaGEdlvQDs1G2slMx6hqSUj7tJtpBh1WqBOHMxw0o5G+FQSbAiudGeZM4Xk+h4QWOlhBh6f2kDCLhDaaQUXuI+EeEjbTDAt5nwU6OjQnOd4D/ZVisRiPp/NvssS4+zFot5igYw9N0lmCVIbDPbPrQEpKCh1istLBCloCHcYoJOEpmkHHFQtJSOeMtdKMBvpfYh8I1pi5d4tLdtBrb520fge1MUeMj6GnPSYm5pN4g3QWr/SPg2Dt2TRNhF57H0kJeoN9R4xH09NeEhPrSfykdFaKsy41W8kfB8FGJ5mbqOTtINjoKBTc0lQBgg2NQsHkMekzFZ7DtQAEu1Es+PP3hMAn6QEIdqNY8Ha+UJ2aqA0IdgOCQXDvgGBjo1jwJx8a9DdNINiNtbBCWQEw2MA4IJhxqoQvGw25VBcIVodyuvTmcb1r4QMQ7KZkpcKNAXbyWSXuSQFGAwS7Sebud48rpZF4gTTIRPumdksJOm9jjRh/mb5FGpeiOwrYp40dm1cPgo3LPnFkmC5OnikNEx8gifUB95+gI8t0/4lNINjAtH3qmdtBB5PrpIkedF5Wx9wQOhVrnxhPoW9J7jY35E/cxnqhvNZ4vwMHweowgUv8UpyrZyxAsDpM4pIdJ6qEY3rXowcgWB2IYKfzGAhmFhDMOOuWfkEFlxjuSxgEq0MSX+50njDgOnkgWB3cgp1WEMwqHsF2EMwqa1bRPhIQzCzjaCsaBLPLJBDMNpO4LSdAMMNM4qKS6xwgmFmIYJ7PtQulx92c1Ls+HYBgdZjELU3Ip6tNe1CyG7K6gGB18DSy2uprKdWCAIIZwyNYpAUEM0dBdpcplSCYSUp2bt7q2UGRCK6TyQmtawWCVaNE3J1NXD5LJqrsFdMLIFg9Goqyyp1fkUZWvc1isQjFxcXVJDxCwhISWg6TyBES1pJQoBmlxcX76UaMzbW12m35AILVISlO7OLIJ9fxLhrZQiI0j17YH9CMHSRCd+PrtkNjeotnh8YvtaoYCFaHjlZ0SUJCAp0+70wkEbrPQAUJP6IZKSRSSsJaErr39NmTkLAlgwi2JGzkj2hVMRCsDt0ek+QjCNX0wXlvWkm9Ru1uEKwOIQrusg2XRnvUg2B1CFGw/bjIYcGudpU8gGB1CFGwxLGMTBBsaBQKzuP5arWq0h0QrA4KBVds5BUuEuIPEKwOCgWTp2YQbGjsVmW/HD2Uo9FalyDYGJRBK5ptQDDjgGCD8+a8PEXvz0jV6JenIFgdoBXNOJO4kREREU/Q6IIID/+gifFi4k0SPybGI3aTRKaUoIs1bYj442P52lQMBKuDZ4Wl39HoNHF9Jff6tQ+IiaUkXiMtyRRPEmlSgop9k4SJ2lQMBKuDfX82hUaLsj24L8kcMUGH/pvEeDb9uj0uJegc+ars4Qpv8X7pRXDaksV2Gh6Njp4ThdvHR0eXgmCteJjbq03B/gW3RrnqXhfjSSm4PRquYA3RQXDeNowjxfgcB26fuOCt0yBYK3QQvDcJ42meqHURxq7TOO1dEp+J0LPaVOWcRt8rOP4zd/D1fPLSYrG8oE1Vzml0eA52RLmOxmKXk0RnkXtzuwvnrYJbtFbo0dGRFhNjx2dmkGb0v0iqct7CmCYQrBXQk8U4IJhxIv+ao03BINgY6NCKBsF9CQhmHBDMOA9yEfM12VgLBBuDRzmO02T/UhBsDNa+NDqiRIuCQbAxgOdgxvkwHn6bxDQwbZZxQDDjlAnjoJHFMmXCA1yBFgWDYGMQv2kECGaZLTwIZhoQzDggmHEOZEEji2mgFc04ZcILM8q0KBgEG4MyYTNv0aJgEGwMyoRNvCa/8QfBxqCGCK7RomAQbBAc/HpNBgxBsFEYzuVqUSwINgogmHFAMOOAYMa5V5up7+eQ4PdvOP9/4qRE8/yhA6+d3eiOx6E/6FapTu7l3sxwqF/suSM4LWzBoaX9dompFy/aWpJw6UwaLbrybmMIXravUf1iGRL88KTJVw9+1u+/0cj76Dn3iKl7nyQvT9OFyxpv3TDBCIKHc7tbnenTps05HvjcIGBJcP83HDW/nimmovuLvCVmXLyEvKwYKO4VOP/KQ86iaxeR2HPjnIYQPGrRMadzE8dxH6paLEuCf0NeEgaKu6vXHRIRN42zo7XkdTMSO3xb5/Y7D/2dRD643mYMwZ9mnXI6D8eP4XaqWixLgunikOXIz7K+drSOvL6P6j3JDVdsOLjp8iXOqsuznMYQXCrUWsjtZSoI9sfDT5GXYnTQk1qARFaLhz236PPFW/Tli8lL7MDWBETu4v1Q/0/6vr5elAob+QYQ3AsPDyVPGeukW/TRXBFplLVbI6ttENW9fGCLjd7FH7vt0EkdKtydvhFcbWbBF00VProy0t/hjsekuHtJ6qmfbS2Jv2KM55BBbtF9IRjdubbVtIKnzLjkosn+93h8/4YBv6IdHYv6kxfbzOsGXjtd3GDdIII37Wik+/Oo+0N/L8EvXIsG/Dn5B3MKfl7vGiiiVCgv1aBY7+9gV8FIhK5OB8F9Tqk2G8x6CW5e/RsUPvX+K0Bwn0MFN9TW1qlcrJfgAeimt504p/dGtUEFmxwiOIPn+VlTJriZSvMWeOIT6I4PzoliYhOJV4rxCZkkkUEjE3f4LtZL8FPZvaoFwdpBBFduT0iYKO7VMYLmSQn3velOMbGCxAVpSw+608NOd2yC72K9BM+vwLg+DgTrQKlQVU0oik9Opbh7XrJTPbjXsUwTE/R34jYxnnqUJGpIuJwb77tY78ekJIzjA/V6gGAtKBdEQtqGNF2W4C0j0M0jRlwzGATrQFODm5xt5IuY9qtlkjCDhI0k5OkQ4gESppKwjWbUksghErp3W9rMb9g67w3fxXYTvNTdeXvpchCsH6l8KIL57f7K87pFD80IIBcEa0zVwRqbjU7dabTZbPTJuI2ENjpE0iRmOKWMZhK61+b5qmZPtr/yugmOTIl0A4L144jQoGp53QQjcYwNBOtHiIKbrX7Gw7oJrmmucQOC9eOIUH+SEmS/pbxWdGQk3KL15oj0tPRlEG9yNKXIEixNggDB+tFQ7uawUB/Em2r4V2QJrqmBW7RBOKaJYGhFGwZtBEMr2jBoIxha0YYhOMHNlg9lDjb8tHny5K1nQbDuBCdY7mMSxs/TOzR8B+vPMaHKIptWp7MibpfvgrwED36suvrRS0Cw7tQLQdDbxe4leMRqjN/8EwjWneY62VQIddacfH8FefVkjQ4fPy58DAg2E/VCXY3M4ULoyTIjbsHvZR/yfRR6skyPW7C852CZgGBDQQRbUlbLFFx23yUDBw4EwWaCCJb/HHz3ZWE3hN0Igs1EvVBaWSm3Jys8+aK69NEg2Ewcp4/C73CjfB/1Enxh1rWvvXExCDYTbV/Z6oU1MgXf8vYkhO4BwSbDLltw2zenV/6rCQSbDPmCYTTJlNiF/R/5WekSRpNYwC743T8cRpNYwC6U+1vHFEaTWMBOHpT8LA0Ao0ks0Hw4+eUVvg/BaBITtG2Q1YqG0STTIk8wwZqQYAvgFwQbELmCPw1HaFAmCDYd73P3zs/0dcBL8G//e/Xq628HwabjY47jpvg64D3YsBnjjeEg2HQ0rJvJPevrgJfgG+9MSeFuAsGmwyYUV/ucPesl+IN+CIVtB8GmwyZU+j7g3YouWrZMCOAXBBuQw1sSfR/oLvinoRsC2QXBhqSIT/B9wOsKvmsJCDYlRfzKpbyvA16CR/V/vOMH4GlLFttp2D4+Orq0MwmCjUgRv0hWK7pLX3RrlKvudbfg6G5JEGxE6j5dJUtwl77ovG0Yu6/l9okL3jrdmQTBRsQm8LIEd5myszcJ42k04jqN097tSG6fPn16X9ceCIhcwV2m7HS5ZPHX8zuSny1fPqOvaw8ERK7gwVOOVT3smbLjiHIdjcUuJ2534bxVYhJu0QblWNZqWYLvScV4pThlJy0mxo7PzMCV8xbGNIlJEGxQivgXff94xUvwI0OefuqCMQFWygLBxqOCj7pHjmBZk3ZAsPGw5eT5XtzS52NSgEk7INh4yB5skAMINh4gmHFAMOOQ5+DHfe6sCwuCs4FNeIMb4+sATHxnA3mCYeK7aaGCn/R1AL6D2cAmrPW9PSUso8QGdLBhbI2PA7CMEhvYhA0cd19TzwOwjBIb2ITMsQ8t9bHZEiyjxAaNJUIhzwe+gmEZJbPS+qUswbCMkmmxyhJ8OIBbEGxYrPkrVgf+DkY3v94Agk2JNZfjfOwO7yU46peo3/CNINiEyBOMsTC9H/RFmxF5gps3PHQeGgqCTYg8wf3RZdNzXSDYhFj3c5ylZ7aX4L+k/hDALgg2KHIEw/ayJkaOYNhe1sTIEQzby5oYa+Hmj6Ani2GsQpWvbOjJYgV5gqEny7TIEww9WabFenDha4EHG6Any7TI6+iAnizTAj1ZjCNLMKx0Z16sB+RcwbDSnWmRN5rUdaU7EGwqqOCynisK+1/pDgSbCyr4w/favLP9r3QHgs0FFfwO32MDNO9NOdyAYBNiLS6siwsoGG7RpsUqVNn5gIJjY2Nfumo8CDYhRHCrIAT6DqbsugMEmxB5gw0vLlgw/5bBINiEWA/OnGfvme3jOzhsMQg2IfI6OvLz84saA/gFwYaECs6vCPwdnLcmGwSbETnPwaNuxenkHv06CDYhcgRf8U/8WL/lt1wHgk2IHMED/nV28J14KayyY0ao4LjNvQv+1U3T0GI89VoQbELktKK3nYeuPvn9FX8FwSbEmvdohK1ndvdWtC33W3wm/yQINiFyp83KAAQbERDMOETwqcTEQKNJINi0eIYLe8zZAcGsQBpZj7/D7/XOBsGsQB+TkhNzvbNBMCvIXUYJBJsUEMw4VPAX1bW1XtkgmBXsRHBuHr/RKxsEM8NxEMw2NhDMNqfmzc0DwUzTIm9eNAg2KyCYcUAw07RZjxeDYIahrej85FSvXBDMDPCYxDggmHFAMONQwQUHcrxyQTAzUMHQimYYEMw4IJhxmuLWFxTXH/fKBcEM0SLk8fzWk93yQDBDtAiFW3i+plseCGaIFkFoTk/sPvcdBDNDa3ZmITSyGAZa0YwDghlHP8GfLFWN2BaV/jUYxCM4cXP3qe99IXjC+kS1uKvnkuaAiEfwDh0ekyacCKb0XnkKBPsFBDMOCGYcj+Caiu7724FgZgi6oyNtyWI7DesXLFzYhNvHR0eXgmBDE9y86NYoV517WdKv2/EX63F7dMeREAWfmXfdgKvG+VmseOkWEKyY4ATnbcNY2iIrj8ftExe8dZpEa7KyZgf5dz2CXSPQsCWzf17s2xwaAYIVQwWXC3KHC/cmYTzNEz0z5xR2ncZp75J4LMdNDfLvegTvQ7/9CeP203jHr8OH5eAaNOqBi5/DePtN4bdn4qEIofljBzTjf6CaGvTo/ZdEnhXPA8FyoQP+wbSiO6/gH2OK3OHX8z1HQrtFr0BL8HctLd8JYX9Jv2uIswYN2joCFReH/e79X4bbMtDNSRXZaLXrF3cS9UNSp6At4nkgWC7BPiY5olxHY7HLiV0rskiy3YXzVikTvBSvRCh2Gerfvx/aX4PG4bdQ4jKUglejBM8t+qbfFKC1RPDT+BiaLJ4HguUS9HNwWkyMHZ+ZgfPGxcRsw5XzFsY0KRGcje442/AqFbymsrLimxoUideind0Ev41GDmglgp/CVegZ8TwQLBeP4JSEum65ffcc7HoA/W7xWBQrhHHp/LDTomCS2nj9IBu+9OcJ9dgZjv5MmnHowlWPoK3ieSBYLvqNJomPSadfuOb8oc8ewTtvueDq8T+KgvH2Gy+8/TOM/+9iFIfxVJRMBY8eccn0s+J5IFguugsOTM7wq36ggv1vYAyC/UME7za4YPRfqRgEh4qd494yuODAgOBeeOP5rXo2skCw5tCeLBguZBgQzDggmHGCG2xQW3DPfBCsMnqtkwWC+wgQzDggmHGoYP0m3fXMB8Eqo28rume+pzRpqlavU7ZAsBwMKViaqtX7lC0QLAedBbdZutIgCpamanVM2QLBIUMF6/LjM1Gw0JVKUbA0VUsKQXDoSK3ojMTExDIxry8F13XleIdgz1QtKQTBoSMJ3sXzfIHTebTiKwN8B0tTtaQQBIeOJLimoqLC6nTu5ksNIFiaqiWFIDh0vDo6jCG4Y6qWFILgkDGm4KAAwb3hJThtczkIZgroi2YcEMw4IJhxQDDjeAnOTDwCgpkCHpMYBwQzjjEFdwz0r0WoFgQrwUtw2aETBhDcOdB/fxh6DQQrQedWdFstoZmkbSS0OL0G/PHJsL9ecysIVoLOglt4AhWUS8LdTq8Bf/w22j0bBRhrAMG9orPg1gQC3dangIQZTq8Bf3z3+UfiSRwEh46XYEvtKQN8B0sD/dZ+iHATCFaAIVvR0kD/KhSdkhKBKkBw6BhSsDTQ//t+zRh/iBaB4NAxpuCgAMG9AYIZp7WyEkaTzjVAMOOAYMYBweyyP6UGBLMMtKIZBwQzjlEESwP9MOCvMkLOcUMIlgb6YcBfC/QXLA30w4C/JvShYDsnspbk5UyWBEsD/TDgrwn6C5YG+mHAX20aLI19KthRLWIleU31omBpoB8G/FXHGK1oaaAfBvxVZzed+aa7YGmgHwb8VSeV32QAwcEBgoPAml8AgpkHBDMOCGYcEMw4sCkH44BgxgHBjAOCGadPBOda1GIUCA6SvhC8+W+qMaNRk38FhukLwYCOgGDGAcGMA4IZBwQzDghmHBDMOCCYcUAw44BgxgHBjAOCGQcEMw4IZhwQzDggmHFAMOOAYMYBwYwDghkHBDMOCGacXgSnLVls7xLpSIJgM+FfcGuUq+71zkhHEgSbCv+C87ZhHNkZ6UiCYFPhX/DeJIyndUak5PwhQ/6md6UB+QR9BbdYLC/oXWlAPv4FO6JcR2OxyylGPK9wizYbvbWiY2Ls+MwMMSK+gmCTAc/BjAOCGQcEMw4IZhwQzDggmHFAMOOAYMYBwYwDghkHBDMOCGYcEMw4IJhxQDDjhCJ44twePD+7Z17fMmu63jWYO2WO3jWInNkzLzZ4wT64bb/CAhQTN07vGuCwFr1rMPtVf0dAsAqwLPjVowoLUEzOBr1rgCNP612DnZ/4O6JUMGBwQDDjKBTc+bslnahfsHBhk641IHw52qJvBWpjYpL8HFImuMvvlnTi63b8xXpda0BY8bK+gn+I+s7vMWWCu/xuST/yeJ0rUJUQq6/gsldXLDvu55gywR0/Y9KRM3NO6VyDV/+ts+DPn2u3z/VzzPRX8I8xRfpWAOcnYZ0FFy8nGv7j+5gywV1+t6QTrhVZuv59wo6FSydHfaNnDb79p+vbv/s5prQV3fG7JZ3IGxcTs03XGlB0voJxVszLZX4OwXMw44BgxgHBjAOCGQcEMw7zggchyr01yPfz+ndXRbl+jw7gliGD/fWXnL50iXbV0xzmBb+26B705KKNzQtSfB5ej+pxcdhtZyPR6h7HfhLD2YPPaFhBjWFeMMYL0E6MyRVcif53+OBN0YP+0IR3/Dp8WI774N23kpdIFBl260+ezLbbLhp0Txk5f9R9Q1sfHjRoWCvOQjv0/QRKOKcEX7DsirA/TkHLhLC/pN81xEmOfX/es+TVcTlCuWLmN6s/S77sPnJ+/yVxa9G6Txa1YieaqfNHUMA5Jfgp/CQ6XI+mLkP9+/dD+8kxK3JPLn0HPYnFzNOTfnFR2CXk/McxTke3PLfLhfH5o/X9BEo4pwRH4vHI9iV6dhlaU1lZQbuPm9EcekoKCcTMlei1uhsGYk+jLDOKQ0n4P/30n9cXMuekYCGMS+eHuWfKXR5BX6lgMXMZWvlxmCh476qMJehdfAy9ou8nUMI5KRjvvOWCq8f/SA9OHkLbylSwmNl418WjbxQFf3bzBYPHfos3Ib1HJBVwDgjulUPo44Dn/OnOPqiIVpzrgvHy+EBnnJmb3xcV0YhzXjDrgGDGAcGMA4IZBwQzzv8DHvtXfb1/4nwAAAAASUVORK5CYII=",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "surv_object_LIDs_rs6269 <- Surv(time = lids$Disease_duration_fromBL, event = lids$dyskinesia_status_at_censoring)\n",
    "\n",
    "fit_rs6269 <- survfit(surv_object_LIDs_rs6269 ~ lids$`chr22:19962429:A:G_G`, data = lids)\n",
    "\n",
    "rs6269_plot<-ggsurvplot(fit_rs6269,\n",
    "                        data = lids,\n",
    "                        pval = TRUE,\n",
    "                        title=\"rs6269\",\n",
    "                        linetype = \"strata\",\n",
    "                        censor=FALSE,\n",
    "                        palette = \"grey\",\n",
    "                        legend.title = \"Genotype   \",\n",
    "                        pval.coord = c(2, 0.65),\n",
    "                        legend=c(0.2,0.2),\n",
    "                        legend.labs = c(\"GG\",\"GA\",\"AA\"),\n",
    "                        xlab= \"Time (Years)\",\n",
    "                        font.x=c(10,\"bold\",\"black\"),\n",
    "                        font.y=c(10,\"bold\",\"black\"),\n",
    "                        font.caption=c(10,\"bold\",\"black\"),\n",
    "                        font.legend = c(10, \"bold\"), \n",
    "                        ggtheme = theme(\n",
    "                         panel.background = element_rect(fill='#efefef'), #transparent panel bg\n",
    "                         plot.background = element_rect(fill='transparent', color=NA), #transparent plot bg\n",
    "                         panel.grid.major = element_blank(), #remove major gridlines\n",
    "                         panel.grid.minor = element_blank(), #remove minor gridlines\n",
    "                         legend.background = element_rect(fill='transparent'), #transparent legend bg\n",
    "                         legend.box.background = element_rect(fill='white'), #transparent legend panel\n",
    "                         panel.border = element_rect(linetype=1, fill = NA),\n",
    "                       ))\n",
    "\n",
    "\n",
    "save(rs6269_plot, file = \"rs6269_plot.rdata\")\n",
    "rs6269_plot\n",
    "#summary(fit_rs6269)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAAC/VBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///9KoeSIAAAdrUlEQVR4nO2deXwURb7Ai0RuRXH3KboHu2+fF7qrtuyyHoA+FkVZXBXBRVBBIZGHnBuzgZCIqAGD8eBIw0ICJlwJJCQghJtgTDgacgy5M4FkSMgxyaiwxoOd+ryqnu6QTGaSmenp6Z7K7/tH1zXTqeHL9NTRXYUwwDRI6woA6gKCGQcEMw4IZhxFgseifeRoGN1v4AsYW6bd2v9PR0k66ra+Y2u9UjlAOe4L/r41tn40FVx509SDx3mMp95x5NyM65twXJ+EnGEPe7OOgALcE8y9M+b6ldvu6n1rMElc+NUFKviVcbaye6IwbkY5+IEFGJ9Dp71fVcAT3BQ88LDV2HPzxdw4jK0j43+ign8W/tjNw4nP1x9uuPrR7Ze/D/iCvPDWVepUF3AXNwXPxDivV5MYjxmDqeDLqP9nJ6fe2ID//RwKHJSHa9ApUnhfhAp1BTzATcErMb46YuDLW77HpYMuioK/QS9j/OOgNXj53fvPBg9uvihenUGwXnBT8L/I4T+H3/4192NCj8DAQBQwydpvKcl7JOz7Xunksv2L1XCJ1hceCCaYUd7XBoMhD/EmPPIV8q2+PfZKjz2iYGhk6Qv3BZ9aln/xs77NNCk2snb1Wl8wc2AjHnHvEcOs3qV4Q5/EE9BN0g3uCy4Z9bO+D2aISVEw5n/T71EB45pJt/QfStMfDOoDAx26AYYqGQcEMw4IZhwQzDggmHFAMOOAYMYBwYzjluBvmgH/4Tv3Bf81EvAbIt52X/BcC+A3mEEw24BgxgHBjAOCGQcEMw4IZhwQzDggmHFAMOOAYMYBwYwDghkHBDMOCGYc54Kvhkw8ZIulRSyulY8g2M9wLtjatM0m2BxiLftAOoJgf6OzS7QkOGsTxkHSEeMzSUkzO55mx+OEpwifbdy4cQkJZzf7/rMADnBBcEYyxtOlI8afjhr1RsfTJHASy3meD6ERo+8/C+AAt7/Bji/RJVsJewmn8vPzs/fu4Lhi338WwAFdCbZacFOItTRKOrr4G3yB49bQk3+4bNnyaEJ6fX29kYTLlxFOkZKcZbaSD0vr6xv2kUhsg+oftXvSieBl02euxVeCSfs5PLxWProm+OJD3ASDwbDvj9KF+zVy4d4mX8V5UhImJ6J4/l9iYn2NDz5tN0SdfnBT6OQIQRB2vzJ58mvzCKFE8E4SBk+ePPnVJFISRSLBJGPmR0Tw2nnzHuVWlPrg03ZDVBroqKusMplMVZWVledJaDpPItX2GfQV1VLGM9yyEoulsUiCfpsvyYl6kqiWE/TkRileThMlra8BHKGTkayx3EwiuFC+cCeQrFQ5cZIkouUE7X0FSfGJ9J1PiNGXvV8lRtCJ4LcemplXUpIpa1xTUlKyUU7sI4klcqKYJF6T4i+QeMlIMTq8AvrdjtGJ4Kq8E4Ln7OSGC3XerxQT6ESwpYn+KntKzqzZwiUVKsUCehHcXNak5O25INgJehFs4jOVvB0EO0MvgrP4nVRRdUFBQQlNnyMR2lkykVDsGxWTiImEtSQ8RzNKSKSKhHUkzD1wHAQ7Ri+Cv+J5Kjab5/kkmo4jkXwSCiSknSbLZhIRSFhAwg00I5lEviJhKc/Hvs8NK1ChUiygF8Hnk5MqSJCblJS0j6ZTSITOV5wjYTrN2E0iVGIJCXfSjAwSOUNCY1JSwl84bud5FWrFAHoRrJBR3EdrYTDbEcwIXsoXaV0JXcKQ4C+1roQuYUbwusSTlgaj0VhJk9UkUktCMwmNdBDTRMIaOYP2uC+SkDbKm2iGWcOaqw0zgg+RYxFpWK+jyZ0kcpyEFSTkG0lkDwnpK6ppxkUSOUDCvSSsoxmV2lVcdZgRvMNgMHy5fv36DSQ0JJHIXhLmkHB9HomkkHA3CU/TDIFE0kiYSsJcmnHCcPrQMYMSihq1/hdwBiOCj2YomawQhAP854reL1Rp/S/gDEYEN5qUcYrfquTtBSBY5+Tbxs88pRAE6xzjumNK3g6C1ebDyHPKTlBXbyk84PG7C4X8EkXTnerBiOAnuRSlI5UNW3iPm8LFpJllUvj3VYIZwdxjCnuzpEd8wtP3mqvyQbCq/JPjuFPKTtG4lk/z/N3nQLDKjFAq2HIu86zFLAgXPHszCFYZUXCNvAz2YZKTKycMJLFfTtBb5BOkeAx94wox+k4Wjddn8nyiR38eBKvNqInkN7hSvn36XyTngJygPaBVcoK2xeZK8WfpG5+xxV+h8ebDyXw8Cc8bjUb6P6GOhOI3+gKJ0Btz60ko3llQRSL0LqFGOllhAcHqcyCFHGqjJehMQ4GcoBPFR+QE9ZYsxdfSN8bS2FvcJNtpqpJSyXEDz/P09hF6w5D4jU6QbhjKJ2EczUgikRwSltDJCgsI1juFCbvbJhPi4uIKSXiWhNtpxjYSySXhORKKt4ilkAh9pqaMhNQ4CGaaamMuCGaZVP4QCGaZVH4rCNY1ZbsOK3l7OmmCgWA9kyK3oj2j6vgXIFjXKBQMrWi9A4JBcOdk7FQ4Ia0WIFhEseDN/Env1MTbgGARENwtBDtYn6moiM4tmNuu71TnYH2n8yBY36RwdIXVYGmWaQLN+l8psYjEy+S5qI0k8YWcoE8nf2KLDl21PV/L+jsHBIsYHqf3zHkumFsFrWiW+Wz+dhDMNtAPZpyTR3T6iCII9g5judVaV8ExINg7gGDGGctFn829qHUtHACCvQMRLAhlWtfCASDYO7zEbSkFwQwzmUupAMEMA4IZBwQzzqrIL0EwyyTz+RWCcEZ/HSUQ7B2I4GpBEMq1rkcHQLB3IIIt5lIQzCybYk5ZLOUgmFlIKxoEswwIZpzJXPwFEMwwk7mQ9EvlwpkKrStiDwj2DkQwz2cKgu7urQTB3mEyF8ofs1SBYFYRG1mWWhDMKiCYcU4cpndVgmCWOZOauiMVBLML3Xdx8xnbJh36mTgEwV7DnCucPCRv0qGbHedBsHdIjhUvznWmHRs2bDhGBJfTJfDoE21fkJCurFhFM+h6l4dISPdwukQzqkkkMy5uq2q744Jg7zCZmx4ZGbmkSNz19ksiuJQuYkkF7yLhURKepxnU4z4S7idhLc2gez0cJqFqGy+CYO8wzfYU6XJxVz0TESwvQ9t+mz15Vz1650frrno1xsyDnq1S7QIg2Duc+Sg6Ojp47Eox0Qi/wWzjgeCGepV2bQHBKkAEFxa5htyfSuBL1akLCPYq60eKDB9+nHSV3hpu42+03/SclJhF4llSfPhyQWg8Tt8wLhYE+wVr5PU7Cquqqt6S4k+TeJW0cwA3i8SL5Vd9RAQfEWPvgmC/oPKkBP1FLZLiubQkV0rQ/lCT/KpKIriWhM9wS0AwkxDBNHie22FUaQdiEKwprYL3qfUXQLCmSILLi+rU+gsgWFMEocxoQ639aUGwppyRZ59C38pW5y+AYE2pLpMYyW1X5y+AYH0AghkHBDPOE+JdmSrQieC0iMW1NCwNC5sTglsmhoXlgmC1GM1lqHNi54LNIdayD6R4cipuCYNvsIpoIDhrE8ZBUnxOE26ZFPrxZRCsFhoIzkjGeLotWr0IY+tlnLaGxN9E6FV1qtKt0fYbnLBfDL5eQA6NRuNsdarSrYkXN6RWAeeCm0KspVHYaiHRmeTa3GLFWSvgEq0WGgjGaeHhtfhKMGlGv0dShvkLw+tBsFpoIdgpINj7gGDGCXopU50Tg2B9MJr766x6NU4MgvXBCxzHqbI5HgjWB5nrHuVOqHFiEKwP9u8ZAYJZJp4HwUwTzz8/9owaJwbB+gD6wYyzPUGlR8BBsD7IE2rVOTEI1gcgmHHyhAnQyGKZPGEkdJNYJmE99IOZBgY6GAcEMw4IZpxjB6GRxTTQimacPGF2cJ4aJwbB+iBPiOM3pamw2h0I1gd5wmZp7VkvA4L1QZ5wvqTErMKJQbA+yBPKq9XwC4J1ArSiGQcEMw4VvDxehU15QLA+oIIJ6V4/MQjWB3nCP4YSwbFeP3E3Erzhjl7/0/oPuOb+AX3vidGyOu2ht+xsjN5TacnPPu/VE3cfwWkBodmRPbZJqQ2r9maE9FijaY3aQgVXCTUWSwrv3Tt3GBI8evKU2wa86nTZ1lGP0tc83DbrgUmqV8pVWm+6A8FOGR34YVPx3W9KqbBAiY+ljBsiyGF572ujCQ07+nr/J89T2gu+JAhnvLRAOEuC7yOHxN41tlRZtoT0k1aL6J5GcahcenVBILrufQ1q6QRR8M7oY5bULaUWSzndPs07J2ZJML3g5qMsx6W1aBU5bkAVUroh++CiPp/7qGpdIwpewM1JWE03PKxK/VzcDs8LsCR4PDmcRl/ZUqFIQm4q2y7RvdoO+L42xLc17ARJMGU3TdcYvXQjPEuCB5P/+6vkS3TpcQmjVOygkfX3O3xbw04QBR8Jnj59JLfVqye2E1zkz4KvnybsuDXIWXFrNyn2EZKauuFA+pyACB9Wr3NaG1nTZMEViQS6Q95+EtLl4KtoBu0kHCEhXbLlEs0wkciXthc4xE4wGrrS7LeCpwbfeP0U5xvxbrij5+9os3lRIDkE/673wIf00w2+Jri2WlqLRdyflgqm+9Mes7TZn3avtD9tDc2gGxAf5vktzk5sJ3j2L1HPv6X84J+C39C6Bgro+PBZfQmBRipJSDU20gx6U88FEtJ7P8w0g7YpqrL5zc5ObP8bbD0xCqHb0kGwj1H2dKHZVOO0qL3ghpj7UL9pj90Cgn2Mrx4f7Ynu+sSCMztvVOtUsF/TKvjT+WGRlHU08U6kjV0kXi7FI2l76is5QfsIKSQ85OzEdoLHH+5ULQhWi1bBwdLGpNNo4k9SIorEz8pbliaTxDY5QW8ReI+ELzg7sZ3gBQUYV8SCYJ+TJ5RcECO5dK94QgJNSPFoulh4pRSPPk0SJ+VEJUnsjZ7NPeXsxPbdpGSME7oa9QDB3idfEATP16rcxo1yVtROcPwIdM+IEbcPAME+p6bsjOC8C98VrgqOFAdvb1oGgjUg1weCMR68twu5IFgtlAg++socZ0XtBAelBomAYA1QIrgT2glG0hwbCNYAhYLjv3Kc305wcUOxCAjWAGWCdz0+0XFB+0t0EFyiNUOZ4GXc044L2l+iEVyiNUOZ4FXcGMcF7S/RxXCJ1gxlguNdEgytaA1RJjiFG++4AFrRekGZ4HTOlUYWtKI1xBeCMf4pbsqUjVdBsAb4RvAb9AoNv8FaoExwduSnjgvsBA94uqjoqRtBsAYoE1zAb3dcYCd4RAzGHz0BgjXAF4KDgsb1mzih37MgWAN8IRhGsjREmeDTMXGOC2AkSy/4phXtEiBYDXwjOO/RG3v37g2CNcA3gv98c8AdAXeCYA3wjeB+KdeXpY8DwRrgi9kkjPse/OW7H94AgjXAN4KHfDIZoYdBsAbkCmdF8ho8ebergpu/uRz9Xj0I1oBzgkS1J+92VTDMJmlGc52IQfBoXf/iHRmOC2A2SWcUeibYKTCbpDNUFgyzSVqjqmCYTdIeDwXnhL8dl+OoAGaTdIaHgrdyo/kdjgpgNklneCg4nXs2ca+jAvvZpOrERFMXfkGwmngs2LWx6H39EOp/AARrh8qC//DfMTG/uR8Ea0ehkF/kLnWuC+4bh/G6fiBYO4oE9yl3XfCdQ1NTubtAsHY0VrlLkVBGBT/vUiPr8x4IBWwGwf5EhU3wcy50kwinli4VuvALgvWFKPhS0RFXBP80eG1XdkGw3hAFWyy5Ln2Dh0WAYH9DElwtnHNUaid4TOAzrQ+Ap0UsrqVhy8SwsNxrSRCsMyTBTrBfq/LaWLQ5xFr2gSg4rF0SBOsMUfCpyFWOS+0EtxmLztqEsfhdbpkU+vHla0kQrDOkVrTbD4BnJGM8nUasl3Hamtbk5hkzZvj2AwCdIwl+saDUUanzW3bafGXx1wtak/uXLQv29UcAOsOtfvCAqeXnRttu2WkKsZZGYasFt1hx1gopCZdo/VEh5BoM27i/zI13VGon+OFdGEdLt+ykhYfX4ivB2DB/YXi9lATB+qOSDkfHc5zjX2E7wU8OfHl8n2e7WCkLBOuK5hpThXDiH889MsFRqcNuUhc37YBgvVErFDgrcthN6uKmHRCsN1wX7BIgWG/YBNc72tUDBLOAKLiB5x0YBsEsQAVnvDSpS8GwILifQgUncc90KRhufPdTXBUMN777KTbBLxubOhbBbzAL2AS/6qgIllFigVrhTG68S4JhGSW/pP6MIPAuCYZllPyThvNE8PhMc8cSWEaJDUzCttdfc2GgA5ZR8lNMgqGB5x08twbLKLEBEWxey5d3LLATfLYLtyBYpxDBlvzMix0L7OeD7/mgCgT7IVSwQ+wEh/wW9Ri+DgT7HSYXu0kYCzN6wFi0/yEKruX5DoOVdoIb1v7lOjQYBPsdrgoORDfPOG4FwX6Hq4Jf3PVDF3ZBsC5xTTBsL+u3iILNJSXN9gWwvSwbuNaKhu1l/RZXu0kwkuWnmIQ90QmOCmAkiw3EkazG7OzOf4NhJMtvEQW70E2CkSw/hQiuj125GkayWIUINnHcpzCSxSo2wZ8nwUgWo9gEO9hyCVa6YwMXBcNKd/6Kq4LbrnQHgv0IVwXDWLSfYhO8PrGrRhaMRfsprnaT9omAYL+DCG6uLovtUjBcov0UV4cqo6Ki3h40EQT7HaJgsyB0NdlA2fYACPY7XL0v+h+hoQuGDADBfgcRfGn+/NqOBQ5+gwMWg2C/w9V+cE5Ozqm6LvyCYB1iE5xT0PVvcNZnh0Gw/2ESCiq67gePuRenk2v0ByDY77goCMe7FnzLPPx0j2VDfg2C/Y7GgjMuCO753tUBQ3EkrLLjj1QTwbFxnQv+3V3T0WI87Zcg2A+hgrtqRW+6Dt1W8/0tL4FgP6Q666mxpo7Z7VvRpuPf4is5NSDYD3FtazuXAMF6BAQzDhF8aXti4gG7bBDMCkRw43qeX2eXDYJZgTayyguzc+yyQTAruNJNAsF+DAhmHBDMOKJgc0mJXTYIZgVRcBW0opkFBDMOCGac6q/enF8LgtkFxqIZBwQzDghmnOrTJ6s73DQLgtmBtqK5Kam77LJBMCtUn36SGF4NrWhWqRZyvwLBDEMaWeUgmGFsgg9n2mWDYFawCW6wzwbBrEAEX4yO7nq1WRDsp8BAB+OIgs3GSrtsEMwK1UKe0VgKs0nMYhKEEzu3xYJgVmmqKMuGfjDbFINgtiGCL9bb5YFghiCC7f2CYJbQSvCeSK8R1eilfwsmoYKT4trf+u4LwX9fneQthlV475+DPajgLXxxuzyfCL7gztk7ZTwI7gQQzDg1a9eaQTDDNApCc3HBpXZ5IJghGl1bEFwmLWJxLQ0rQhcurMctE8PCckGwrnFPsDnEWiYuS/p1C/5yNW4Jay3xUPCV+b/uOWiCk8WKI+NBsGJMMTFm+zzngrM2YSxvkZXF45ZJoR9fJtHigwdnufl3bYKtI9CDEbN+cdqxOTQCBCuGtqLzhZp2ec4FZyRjPN0WvTLnErZexmlrSDyK46a5+Xdtgg+hP/yEcctlvOXufg9m4mI0ZuQNr2O8+a5+9x/AgxFCC17o2YDfQsXF6KnHbgy6Kr0OBLuOe92ka9/gH8NPieHXC2wlnl2il6MI/F1j43dCwIvpwwZailH/jSPQ6dMBD234bT/TXnRPcsFhFGP91VCifuCuqSheeh0Idh33BDeFWEujsNWCrcsPkmSLFWetUCY4EkcjFLUUBQb2QEeL0QT8MUpailJxDEq0XaLvuu8EWkkEv4zL0RTpdSDYddwc6EgLD6/FV4Jx1oTw8E3YMH9heL0SwYfRA1erllDBnxkMBd8UoyC8Em1tJ/gTNKqnmQgej8+hV6TXgWDXoYJTE8va5fmuH2wdiR5a/AKKEgK4dP7By5Jgklr3m/4mfNMvEiuwpR/6G2nGob4rnkQbpdeBYNfRajZJ6iZdnn17r8GvFuKtQ/rcNvFHSTDefGff+/dj/P4NKBbjaSiFCh434sYZV6XXgWDX0Vhw12QOH/QDFex8A2MQ3Bm6F4z+axcGwZ5Tl5GhyZMNMFTpI+hQpXaNLBCsOlQwTBcyDAhmnMp58xpBMMO4N9ngbcEd80Gwl9G2m9QxHwR7GRDMOETwJPs8EMwQVcO4UdrddNcxHwR7mca9ice0a0V3zJdOtxKhkrYhCPYUbfvBHfOl0z0WgN5tG4JgT9FYcLOxLVWy4JqAl26/t00Igj2GCtbk4TNJsNAWgyz4E7RzFiq8FoJgj3HzxnevCy5rS6Us+M+9ChNQ5LUQBHuMxoI7IJ6sugci3NUagmDPsQlOT01NLSepwtR9dToQvAKFpaaORQVyCII9xyZ4Pc/z5AfQcprn83Qg+I89GjDejhbJIQj2HJvgs6SFc5GkqjbzuToQ7B4guDPsfoPT4vJBMFPospEFgr0HCGYcEMw4IJhx7AQfSCoEwUxhJ3gndJMYAwQzjk4FSxP9na7RAoJdwU5wXvYFXQi2TfR3vkYLCHYFracLSwh04yYTCY0Wuwn/1jVaQLDHaCy4kSdQQcdJuNNiN+Evr9ECgj1HY8HmRALd1ucECfda7Cb85TVaQLDn2Ak2llzSwW+wPNEvr9ECgj1Hl61oeaJfXqMFBCugqd0D/voQ3DrRL6/RAoK9hT4EuwcIdgMQ3B0AwYwDghkHBDMOCGaXo6nFIJhloBXNOCCYcXQiWJ7ohwl/byNkVupAsDzRDxP+6qC5YHmiHyb81cGHgms5iZUkL3OKJFie6IcJf3XQgWDbRD9M+HudKqNvHwBvKpKoJnn1FZJgeaIfJvy9jj5a0fJEP0z4ex19CG6d6IcJf2+jE8FuAYLdYNcGfdz4DoJV4ny2BQQzDwhmHBDMOLApB+OAYMYBwYwDghnHJ4KPG73FGBDsJr4QHPea1wiuU+VfgWF8IRjQEBDMOCCYcUAw44BgxgHBjAOCGQcEMw4IZhwQzDggmHFAMOOAYMYBwYwDghkHBDMOCGYcEMw4IJhxQDDjgGDGAcGM04ngtIjFtW0irUkQ7E84F2wOsZZ9cC3SmgTBfoVzwVmbMA66FmlNgmC/wrngjGSMp1+LyMkFAwe+pnWlAddx+xvcaDTO1rrSgOs4F9wUYi2NwlaLFLEd4RLtb3TWig4Pr8VXgqWIdATBfgb0gxkHBDMOCGYcEMw4IJhxQDDjgGDGAcGMA4IZBwQzDghmHBDMOCCYcUAw44BgxvFE8KS5HXhjVsc83zJzhtY1mDt1jtY1CHqzY16U+4Id8PujCk+gmNgJWtcABzRqXYNZS5yVgGAvwLLgJaUKT6CYzLVa1wAHdbH5sfps3eOsRKlgQOeAYMZRKPjac0saURG6cGG9pjUgnB9n1LYCJeHhyU6KlAlu89ySRnzdgr9crWkNCMv/qa3gH0K+c1qmTHCb55a0I4vXuALnEqO0FZy3ZPnSSidlygS3PsakIVfmXNK4Bkv+rbHgI6+31M51Uub33+Afw09pWwGck4w1Fnx6GdHwH8dlygS3eW5JI6zLD2r69wlbFkZOCflGyxp8O8/67f85KVPaim59bkkjsiaEh2/StAYUjb/B+GD4P/OcFEE/mHFAMOOAYMYBwYwDghmHecH9EeWRYuS4v/7doBDrH9Ex3DhwgLPxkss3RahXPdVhXvC7ix5Gzy1a1xCa6rB4NarApwN+fzUIxXQo+0kKZw24omIFVYZ5wRiHoq0Yk2+wAT0/fMD6sP5/qsdb7u73YKZY+Od7ySEIBQXc+5Mts/n31/d/OI+8fsyjg82j+/d/0IwPoi3afgIldCvBfZbeEvD4VLRUCHgxfdhACyn7/rpXybHp5wgdlzK/idmfcvOj5PWBEbEr0ao9i8zYgt7U+CMooFsJHo+fQ2cr0LSlKDCwBzpKyqqReHPpp+g5LGVenvyr6wNuJK9/BuN0NOT1bVaMe43T9hMooVsJDsITkek8enUp+sxgKKDDxw1oDn1JKgmkzGj0btkdvbGtUXYghEPJ+D89tL+vz2O6pWAhgEvnHxTvlPv5WHqkgqXMpSh6d4AkOGPF3gi0Bpejd7T9BEroloLx1iF9bpv4Iy2cMpC2lalgKbNu2A3j7pQE77+nz4AXvsXrkdYzkgroBoI7JRvt7vI1Twz1QUXUorsLxssSunrFlbk5vqiISnR7wawDghkHBDMOCGYcEMw4/w+FR4dj7a8KBwAAAABJRU5ErkJggg==",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "surv_object_LIDs_rs4680 <- Surv(time = lids$Disease_duration_fromBL, event = lids$dyskinesia_status_at_censoring)\n",
    "\n",
    "fit_rs4680 <- survfit(surv_object_LIDs_rs4680 ~ lids$`chr22:19963748:G:A_G`, data = lids)\n",
    "\n",
    "rs4680_plot<-ggsurvplot(fit_rs4680, \n",
    "                        data = lids, \n",
    "                        pval = TRUE,\n",
    "                        title=\"rs4680\",\n",
    "                        linetype = \"strata\",\n",
    "                        censor=FALSE,\n",
    "                        palette = \"grey\",\n",
    "                        legend.title = \"Genotype   \",\n",
    "                        pval.coord = c(2, 0.65),\n",
    "                        legend=c(0.2,0.2),\n",
    "                        legend.labs = c(\"AA\",\"AG\",\"GG\"),\n",
    "                        xlab= \"Time (Years)\",\n",
    "                        font.x=c(10,\"bold\",\"black\"),\n",
    "                        font.y=c(10,\"bold\",\"black\"),\n",
    "                        font.legend = c(10, \"bold\"), \n",
    "                        font.caption=c(10,\"bold\",\"black\"),\n",
    "                        ggtheme = theme(\n",
    "                         panel.background = element_rect(fill='#efefef'), #transparent panel bg\n",
    "                         plot.background = element_rect(fill='transparent', color=NA), #transparent plot bg\n",
    "                         panel.grid.major = element_blank(), #remove major gridlines\n",
    "                         panel.grid.minor = element_blank(), #remove minor gridlines\n",
    "                         legend.background = element_rect(fill='transparent'), #transparent legend bg\n",
    "                         legend.box.background = element_rect(fill='white'), #transparent legend panel\n",
    "                         panel.border = element_rect(linetype=1, fill = NA),\n",
    "                       ))\n",
    "\n",
    "\n",
    "save(rs4680_plot, file = \"rs4680_plot.rdata\")\n",
    "rs4680_plot\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAADAFBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////isF19AAAddElEQVR4nO2deWAU1f3AH6GABOWwrdVWiz38gdralpFKEQ21FKUiVkVQDCAoJFLlbMRASESsAYNRuTKcARKuBMjBFe4rJBwDSTaQiySQbLLk2GTFUONB9/3em53NsdlNdjM7O7Mv388f833H7uQtH3b2HXMgDDANUrsBgLKAYMYBwYwDghlHluCR6ADZZg/37fMKxpEPd//xi1cbI6AJXBf8bUNq3XAquLj3pMOneJLbln5gSL/GCGgC1wRzH464c/n2/t1+Fkgy1x+4TgVPGNVYfwYZm0VAdVwU3OeouajLlrKMDRibh0b/QAX/OOSpu5++IFbXTnnE3DQC6uOi4GkYZ3atEdORIzAVXId6LDs3qVcVxrs6o/7XcWMEtICLgpdjfNuvzxtbv8X595aJgm+iNzD+/t5VGN/MPvS3IbcbI6AFXBS8lmz+d/T9X3Lfx3Tq3Lkz8hln9l1Eyp4MFutNnY42i4DqtEMwwYgyv8rOzs5EvB4PnUC+1T+PEsur0IlmEVAd1wWfX5xVtqx7Lc2KnazErut00/pU48mJZxP+8tv/NkRAE7guOG/Yj7sPSBGzomDMP+g7RMB4/ANd7/cvboyAJoCpSsYBwYwDghkHBDMOCGYcEMw4IJhxQDDjuCT4Zi3gPXzjuuAXwgCvIfR91wXPNAFegxEEsw0IZhwQzDggmHFAMOOAYMYBwYwDghkHBDMOCGYcEMw4IJhxQDDjgGDGcSz4dtDYI5ZUUugCg3ULgr0Mx4LNNdstgo1B5oJPpC0I9jZaO0RLglM3YRwgbTG+GBc3reVudv6V8Bxh2caNGxeSOL3W858FsIMTglPiMZ4ibTH+ctiwt1vuJoaTWMLzfBBNFHn+swB2cPkbbP8QnbeNsJ9wPisrK23/To7L9fxnAezQlmCzCdcEmfPDpa2Tv8HXOW4V3fmnixcvjiAkV1ZWFpG4hOQXnyc16SQuIQWf5ldWVh0giagqxT9qx6QVwYunTFuNbwWS/nNIiMG6dU5w2ePcq+SrfODP0oF7Ejlwb7UexaNITbA1E87za8TMujIPfNoOiDLj4JoP/BcIgrB3gr//xFmED4jgnSS+4+/vPyGO1CwmiUBSMO0zInj1rFlDuKX5Hvi0HRCFJjqqrpeVl5frrxNILC8h0VpQQgtohZ7EMqngeW5xnslUnSNRTnZxw5qpJJlSa4buvEhKX6WZvIbXAPbQyEzWSG4aEXzFeuCOIUUJ1sw5komwZujoK0BKj6XvfEZMvuH+JjGCRgS/9/i0jJycE1aNK8iXMtqa2UcyH1ozV0jmTSn9Mv3uDhWTTxfAuNs+GhFcknVWaD+7uKeFG+5vFBNoRLCpprSk/Zx/bzoIdoBWBJuuGOW8OwMEO0ArgstWH5fzdhDsCK0ITuXj9SSUkF/ULJrPIokSEstIvEQLskmimEQD/dGlBbkkFpJYQeK5Q6dAsH20IvgMz+tIEHiej6X5WJK4QKKOxPW0II4k0kjMI5GnXeZEEk+QeI3no/7DDdIp0CgW0Irg0qSEAhJyEhISUmg+hSTorEYBiXtowRGSoN/tayQmUMEnSaRfZX1Cwva/c9wumAmzi1YEy2QY9wWvV7sRmoQZwYt4OEjbgyHBh9VuhCZhRnB0XLqpOo9QQ7LXSaR9cCMtoAPsEhKvkVhLC+jSs55E8aQTWlChYsuVhhnBR0xih5rnqa39JB4isZwWUNNHSaR9NSMtoIOr0yTuou9cSxJX1Gu44jAjeBsZDp+OJpwjiQQSE0lMowVnSCKZxF0knqcFp0hiH4l0YVrYSBLHhDMHDsuYCydDdc0eBBgRfDpFzmKFIBziN8t6vzgpo0kYEVxbJY9L/A45b78MgjVOFh8v5+1XQLDGKV2/X87bQbDSfBp2WfY+ChPa/dYrpJul0fN+GRH8LLe7XOYuKjfy7XZ0lXSzNDpTyoxg7qliebuosIyU20e1DgQrygccx52XtwvjGn5L+999GQQrjJ9cwaZrulxT9eHDOe16MwhWGlFwufU22EdJSYY1k00yB60Zeop8jJSOpG9cKiY/TBX3Qo7TG9v150Gw0gwbS36Di62nT68lJYesGXrexwprhvbFZkrpF+kbn7ekJ4h7Mabw0STk63Q6OvCpIFFHzy64SiL9ja+iBXT1oohEOqdtpAVVIFh5Du0mG0OExCmS0Vkz9KB7zJqh3+B4Kb2avjGKpt7jxll2U7R6H9nu5Hn+NIlX6doE9bmHRHpUKKEFBpI4SOIBEm/QgmsgWOtciZF60EbxhKFDcXFx9HSgayTGUcHHSUwnsYwW0PP7TpNI/xdV0oJSEMw4pUUZIJhlEvgjIJhlEviNqSBYyxQkHpXz9v08fwoEa5nd1l50+6i+fOIcCNYyMgVDL1rrgGAQ3Dopu+QvSCsCCBaRLXgLf849LXE3IFgEBHcIwXbuz5STQy+IMDa9v1OFnfs7XQPB2mY3R++wGiitMo2hRX+TMvNJusC6FkXXEvdZM2dI5gtLcuCKHVlqtt8xIFgk+6/0kqb2C+ZWQC+aZZbN3gGC2QbGwYxz7pjMkzqVAgS7h5HcSrWbYB8Q7B5Gcp8XXNXixQ0g2D2M5CIEoUDtVtgBBLuHkdyybBDMMK9zcYUgmGH8ud0gmGVAMOOAYMZZEXa6UMi4rL173oJg9xDPZ10TLPc31hYg2D0QwbVlOcJVtdvRAhDsHohgeicHEMwqKyNAMNO8xu0GwSzjD4LZxp+Lvk4EZ1xTuyG2gGD34M8FbT9XID0yRkuAYPdABPP8sWsgmFX8uU/3ZZoMIJhVxE4WCGYXEMw4Z48Wm4yVJedAMLtc3MLzfFyWSJ7ajWkABLsN3Wqejz4qPcNBMw8kB8Huw1hZWVZeSL6/Oiq4IC8vjy4PG0gUzwQoJAnx6kT6qCZaUEQivR6iyvKQJ4UaBYLdQnyU9Ot7lhynt1HBG0kikxRcInETrdlKEvSOuFdIpPfSbHbDRMWuiwDB7sGfmxIWFrYwx3QxNjY2kQqOJwl6WwcdieITPxJJIoPEPBK30oL9JEGvKi6KXaPcs7lAsHuYbLmKdImYqXb1N/j04etKNIoCgt3Dxc8iIiICRy4XMy4LVhAQrABE8CUnyZb+J1RV1ijTFhCsADUZzj8Tr9rylhheoSeYg2C3sm6oBb/CioqKuX4WxpB0xRgp8z5JF0tpP54IPkVfPyoKBHsFq6z37yglmdlSeiSteUHKzCLpMuurVhLBx8TURyDYKyg+J0F/UXOkdAatyZAy9PkCNdZXFRPBBhKf5xaCYCaRfoNf5nYWVSvzF0CwqjQIPqDUXwDBqiIImXTx6cSRC1dgmMQiTcZTBmX+AghWlepyiQ/eS1XmL4BgbTCU26HMjkGwNgDBjAOCGecZ8axMBWhFcFLoAgON+cHBM4Jw/djg4AwQrBTDuRRlduxYsDHIXPCJlI5PwPXB8A1WEBUEp27COEBKz6jB9ePmfl4HgpVCBcEp8RhPsSRL52NsrsNJq0j6HYQmKtOUDo263+CYg2L4ag7ZVBcVTVemKR2aaF6nzI4dC64JMueHY7OJJKeRY3O9GacuhUO0UqggGCeFhBjwrUDSjf6Y5LJnzwupBMFKoYZgh4Bg9wOCGSfgtZPK7BgEa4Ph3AvvViqxYxCsDV7hOE6Rh+OBYG1wcs0Q7qwSOwbB2uDgXj8QzDLRPAhmmmj+5ZEXldgxCNYGMA5mnB0xOcrsGARrg0w4bZZtQDDjZApjoJPFMpnCUBgmsUzMOhgHMw1MdDAOCGYcEMw4sXuhk8U059NBMNNkCtMDz1YpsGMQrA0yhQ08v16Bk3ZAsDbIFPbwPK/AbaNBsDaAuWjGyRSulhqV2DEI1gYwF804IJhxqOAl0Qo8dQkEawMqmJDs9h2DYG2QKfx7IBEc5fYddyDB6x/q+tuGf8DIwb17DhJvbJM//qfd/m+Les2yQIdJ8RGbik1Zae59xnTHEZzkMzctrNN2KffEvJ17XuhywmQq/fUTmw9v3qdq00wWwTVlFSbTbt69Z+4wJHi4//j7ek6scFQ9bAh9zeAmJdW9g02m2X0dvsOTNEx0gGCHDO/8aU3uw+9IueDOEp9LBXeFks2Sbk1mEyru+sxk6jd+9E/6z1Xkyk1XsBF8YN9+N01bsiT492QT263ckitIk5B+0gyIPtNoA7ra+IZp9xSbTN26Tj2xtneApxtriyh4V8QJU3IsNbuW50+4Z8csCR5HNlnIwW15DWgF2a5HhQ0loT2PkG3XfrUm04ddFbqhvtOIgudwM2LWb6B3Bj8bzx93z45ZEjyabC6gM5bcXCQRKVVbDtFdGw7Rc/oco+GBUWSzDV32bFtbIAmm7KH58iI3LT6wJLgv+b+/wnqIzj8lUSRVN+9k1Qb81PJVH/0w2XykjW/wscApU4Zy26SiMr1eT/87VpFosBbQdlaTKH7KcpKgJwkY9WUOHwhgIzjHmwXfOVnY+TOHv6YNw6SoJ0luQrd1RH62yXSiy3vnNt/9Lw+20y4NnazJDYLX8Dyfa7I8sFYc3bV4YO0W6YG1l3l+j6Md2whGA5cbvVbwpMBed46/4bB+/UNdfkMnOuZ3Jhtf8fD9OkntfKzbgxrqRRtKrW1xSbDDmRobwdPvR13+ufs77xT8ttotkIG8Bf9cpwVj89lhCN2XDII9jDzBRn25w6rmgqsif498Jz91Dwj2MJ46ZacL6v+FCZ9svVOtUcFeTaYgZIojuC9nB4dR1tDMh2EWEkn6qpQOSyOZM9YMHSPsJvGIox3bCB59tFW1IFgpcgRBEPuHgdKDSSfTzBNSJpykL1kfWRpPMtutGXqKwMckvuJoxzaC5+gwLowCwR6ntvqSRXAGfVY8IYZmpHQEvVl4sZSOuEAy56yZYpLZHzGde87Rjm2HSfEYx7Q16wGClSBDcDzCa4vt3DBHVc0ER/uhR/z8ft4TBKuAJwSHiaP/3otBsAp4QjDGffe3IRcEK4UcwccnzHBU1UxwQEKACAhWATmCW6GZYCStsYFgFZApOPqM/fJmgnOrckVAsArIE5zsN9p+RfNDdAAcolVDnuAoboT9iuaHaASHaNWQJ3iDU4Jzc+EQrRryBEc7JRh60SoiT/BuzpnfYOhFq4jMThY31n4F9KK1gicEY/zDhvHjN94GwSrgGcFv0yM0/AargTzBaWFf2q+wEdzzHzk5z/UCwSogT7CO32G/wkawXyTGnz0DglXAE4IDAkb5jh3j+yIIVgFPCIaZLBWRJ/hC5Ab7FTCTpRU804t2ChCsBJ4RnDmkV7du3UCwCnhG8F/u9nnIpx8IVgHPCPbdfWdB8igQrAKeWE3CuPvh+z/69C4QrAKeEfzoF/4IDQbBKuAZwbU36yI+rgTBKuAZwbCapBryBOfuTLFfAatJWsET50XDapKKeEYwrCapRoZwOYeS175HN1QXXbN7KyVYTdIKmYJEaXvenR4ymz9orwJWk7RCZYmITmjXXUjjuWf5nfYqYDVJY1xpn+Bk7uXY/fYqbFeTSmNj9W34BcFK0m7Bzs1FH/BFqMchEKweCgv+w68jIx/8IwhWD4UFd9+A8RpfEKweCgvuNzAhgesPgtVD4U7W5k4I+WwBwerRbsEvOTFMIpxftEhowy8IVpJ2Cr6Rc8wZwT/0Xd2WXRCsLFeErBxXER8MlOHUN3hQKAhWl1zBdQroG0sFu8+dsBE8ovPzDReAJ4UuMNBYPzY4OKMxC4IVxVjiKjkWwY72Z3Ovysa5aGOQueATUXBwsywI1hiFVPD5sBX2a20EN5mLTt2Esfhdrh839/O6xiwI1hii4HZcAJ4Sj/EUmjDX4aRVDdktU6dO9ewHAFpHEvyqLt9ereNTdpp8ZfFXcxqyBxcvDvT0RwBaQxLs3Di456Srl4dbTtmpCTLnh2OzCdebcepSKQuHaO0hCf77zGh7tTaCBydiHCGdspMUEmLAtwJx9ux5IZVSFgRrD1HwXo6z/ytsI/jZPm+MvuPFNu6UBYI1hShY/++XBr9ur9buMKmNk3ZAsKYotIyDy+0/xdzuMKmNk3ZAsKYoFPKqqmod1cIF4F5PIZ2tpA85rLT3BEYQ7PWUXySCdSZTFc/bMQyCWcBABKe8Nq5NwXBDcC+FCo7jnm9TMJz47qU4KxhOfPdSLILfKLJzdRL8BrOARfBEe1VwGyUWcF4w3EbJK3FeMNxGySuxCB590s6lxXAbJRaggo+89aYTEx1wGyWvxCBkFFZU8XxWyyq4jRILGARByDau5c+3rLIRfKkNtyBYk9QW5QjZpms6O3d/sF0PfuSTEhDsheiJYLvYCA76Fer09BoQ7HXoBd6pYRLGwtROMBftfYiCDTzfYrLSRnDV6r//CPUFwV6Hs4I7o7unnjKDYK/DWcGvJn7Xhl0QrEmcEwyPl/VaRMHGvLwWJ9/B42XZwLleNDxe1mtxdpgEM1leil7YGxFjrwJmsthAnMmqTktr/TcYZrK8FlGwE8MkmMnyUojgyqjlK2Emi1WIYD3HfQkzWaxiEbw5DmayGMUiuM31YLjTnbfipGC405234qzgpne6A8FehLOCYS7aS7EIXhfbVicL5qK9FGeHSQdEQLDXQQTXlhZEtSkYDtFeirNTleHh4e/fOxYEex2iYKMgtLXYQNn+JxDsdTh7XvS/586d82hPEOx1EME3Zs82tKyw8xvsswAEex3OjoPT09PPV7ThFwRrEIvgdF3bv8Gpy46CYO/DuXHwiN/hZHKM/gQEex3OCb5nFv5Hp8WP/hIEex3OCe7y8e2eA3EY3GXH+7AIjtrQuuDf9J+CFuDJ94Ngr0MvXBLa7kVv+hG6r/zbe14DwV5HuSCkPjdS37KieS9af+prfCu9HAR7H/p8px5t5xQgWIuIzy6srLItBsGsQATf2MHzu2yKQTArEMHGzTy/xqYYBLNCKelklZSlpdsUg2BWKD3lzGIDCPZaQDDjgGDGEQUb8/JsikEwK4iCS6AXzSwgmHFAMOOUnnlntgEEs0spLDawDQhmHBDMOKUXzpXaeUw0CGYF2ovmxick2hSDYFYovfAsMbwSetGsUipknAHBDEM6WVdBMMNYBB89aVMMglnBIhhOumMWIrgsIqLtu82CYC8FJjoYRxRsLCq2KQbBrCAKhtUkdiGCq84diwLBrALjYMYpFS6eAsEMUy4IRziurNKmGAQzQ7n+EsfZ+gXBLJGrkuC9YW4jvNpN/xZMQgXHbWh+6rsnBL++Ms5dDCp03z8He1DBW/ncZmUeEXzdlb23ymgQ3AogmHHKV682gmCGqRaE2lzdjWZlIJghqp27IbiVpNAFBhoL586bV4nrxwYHZ4BgTeOaYGOQuUC8LelX9fj0Slwf3FDTTsG3Zv+yy71jHNysOCwaBMtGHxlptC1zLDh1E8bWR2Sl8rh+3NzP60gy9/Dhd138uxbBZj80IPTdX1ywbw75gWDZ0F50llDerMyx4JR4jKdYkrdm3MDmOpy0iqTDOW6yi3/XIvgI+sMPGNfX4a0P+w44iXPRiKF3vYXxlv6+fzyE+yKE5rzSpQq/h3Jz0XNP9Qq4Lb0OBDuPa8Okxm/w9yHnxfjVHEtN+w7RS1Ao/qa6+hvB59XkQX1MuajHRj904YLP4+t/5avfjx6J1x1FkeYHBhL1fRInoWjpdSDYeVwTXBNkzg/HZhM2LzlMsvVmnLpUnuAwHIFQ+CLUuXMndDwXjcGfo7hFKAFHoljLIbr/78+i5UTwG/gqGi+9DgQ7j4sTHUkhIQZ8KxCnjgkJ2YSzZ88LqZQj+Cj60+2ShVTwsuxs3c1cFICXo23NBH+BhnUxEsGj8WU0QXodCHYeKjghtqBZmefGweah6PEFr6BwwYdL5gfUSYJJbs2DPfS49y9iC7HJF/2TdONQ96XPoo3S60Cw86i1miQNk+qm/7xr34lX8LZH77hv7PeSYLylX/c/HsT4P3ehKIwno91U8Ci/XlNvS68Dwc6jsuC2Ofn0vd9RwY4fYAyCW0PzgtFPEzEIbj8VKSmqXNkAU5Uegk5VqtfJAsGKQwXDciHDgGDGKZ41qxoEM4xriw3uFtyyHAS7GXWHSS3LQbCbAcGMQwSPsy0DwQxRMogbpt5Jdy3LQbCbqd4fe0K9XnTLcnFnPRDlSSmAYDmoOw5uWS7u7KP5g9FL89dIAQTLQWXBtUVNKWk4RM9F25oEENx+qGBVLj6TBAtNyQbBbsfFE9/dLrigKcUg2O2oLLgFINjNWAQnx8XFkeOjKTcu6QYIZgqL4Bie5wWSy+L5DBDMFBbBOTqdTk9y5Vs1ItgVQHBr2PwGJ23IAsFModFOFgh2FyCYcUAw44BgxrERfCjuCghmChvBu2CYxBggmHE0Kdi60l834xddH4L1YFnYCM5Mu64BwdJKv9kP/Tls/GgQLAe1lwvzCPTBTXoSi0zN56KPoD/cxvhbECwHlQVX8wQq6BSJu0zNBdNbtDgBCG4NlQUbYwn0sT5nSdxvshUcBoLlYiO4KE9D68FHxUN0PQiWgyZ70ZJg81A0MAQ6WfLQsmDxFi2/hWGSLGqKi7Uo2BVAsAuA4I4ACGYcEMw4IJhxQDC7HE/IBcEsA71oxgHBjKMRwdJKPyz4ux3hZLEGBEsr/bDgrxCqC5ZW+mHBXyE8KNjASSwnZSfHS4KllX5Y8FcIDQgOaxJAsBspKarwqOCaHIlSUlZZKAmWVvphwd/9aKMXLa30w4K/+9GGYOtKPyz4ux2NCHYJEOwCiWu1cOI7CFaMkjQTCGYeEMw4IJhx4KEcjAOCGQcEMw4IZhyPCD5V5C5GgGAX8YTgDW+6jcAKRf4VGMYTggEVAcGMA4IZBwQzDghmHBDMOCCYcUAw44BgxgHBjAOCGQcEMw4IZhwQzDggmHFAMOOAYMYBwYwDghkHBDMOCGYcEMw4rQhOCl1gaJJoyIJgb8KxYGOQueCTxkRDFgR7FY4Fp27COKAx0ZAFwV6FY8Ep8RhPaUxYs3P69HlT7UYDzuPyN7i6qGi62o0GnMex4Jogc344NpukhGULh2hvo7VedEiIAd8KlBLSFgR7GTAOZhwQzDggmHFAMOOAYMYBwYwDghkHBDMOCGYcEMw4IJhxQDDjgGDGAcGMA4IZpz2Cx81swdvvtizzLNOmqt2CmZNmqN2CgHdaloW7LtgOjx2XuQPZRI1RuwXYp1rtFry70FENCHYDLAtemC9zB7I5uVrtFuCAOrVbsG2voxq5ggGNA4IZR6bgxuuWVKJw7rx5laq2gHBtVJG6DcgLCYl3UCVPcJPrllTiq3p8eqWqLSAs+UBdwd8FfeOwTp7gJtctqUcqr3IDLseGqys4c+GSRcUO6uQJbriMSUVuzbihcgsW/ldlwcfeqjfMdFDn9d/g70POq9sAnB6PVRZ8YTHR8D/7dfIEN7luSSXMSw6r+vcJW+eFjQ+6qWYLvp5l/vpfDurk9qIbrltSidQxISGbVG0BReVvMD4c8kGmgyoYBzMOCGYcEMw4IJhxQDDjMC+4B6I8mYvsj9e/uTfI/Gd0Alf36elovqSud6hyzVMc5gV/NH8wemn+mqq5CXarV6JCfMHnsdsBKLJF3Q9SfLfnLQUbqDDMC8Z4LtqGMfkGZ6OXn+65LrjHE5V468O+A06KlX/5HdkEoACf3/1gKax97M4egzPJ60cM6Wsc3qPHACM+jLaq+wnk0KEE37HoHp+/TkKLBJ9Xkwf1MZG6b380kWxrfoLQKanwZuTB3XcPIa/vHBq1HK3YO9+ITegdlT+CDDqU4NH4JXSpEE1ehDp37oSOk7pSJJ5c+iV6CUuFdf4P3OnTi7z+eYyT0aNvbTdj3HWUup9ADh1KcAAei/TX0MRFaFl2to5OH1ehGfQlCSRIhRHoo4KHumFLp+xQEIfi8f86qX9eX7vpkIIFHy6ZHyCeKfeTkXRLBUuFi1DEHh9JcMrS/aFoFb6KPlT3E8ihQwrG2x69476x39PK8X1oX5kKlgorBt01qp8k+OAjd/R85Wu8Dqm9IimDDiC4VdLQnjZf88xADzREKTq6YLw4pq1X3JqZ7omGKESHF8w6IJhxQDDjgGDGAcGM8//BxB+C6v2x4AAAAABJRU5ErkJggg==",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "surv_object_LIDs_rs4633 <- Surv(time = lids$Disease_duration_fromBL, event = lids$dyskinesia_status_at_censoring)\n",
    "\n",
    "fit_rs4633 <- survfit(surv_object_LIDs_rs4633 ~ lids$`chr22:19962712:C:T_C`, data = lids)\n",
    "\n",
    "rs4633_plot<-ggsurvplot(fit_rs4633, \n",
    "                        data = lids, \n",
    "                        pval = TRUE,\n",
    "                        title=\"rs4633\",\n",
    "                        linetype = \"strata\",\n",
    "                        censor=FALSE,\n",
    "                        palette = \"grey\",\n",
    "                        legend.title = \"Genotype   \",\n",
    "                        pval.coord = c(2, 0.65),\n",
    "                        legend=c(0.2,0.2),                        \n",
    "                        legend.labs = c(\"TT\",\"TC\",\"CC\"),\n",
    "                        xlab= \"Time (Years)\",\n",
    "                        font.x=c(10,\"bold\",\"black\"),\n",
    "                        font.legend = c(10, \"bold\"), \n",
    "                        font.y=c(10,\"bold\",\"black\"),\n",
    "                        font.caption=c(10,\"bold\",\"black\"),\n",
    "                        ggtheme = theme(\n",
    "                         panel.background = element_rect(fill='#efefef'), #transparent panel bg\n",
    "                         plot.background = element_rect(fill='transparent', color=NA), #transparent plot bg\n",
    "                         panel.grid.major = element_blank(), #remove major gridlines\n",
    "                         panel.grid.minor = element_blank(), #remove minor gridlines\n",
    "                         legend.background = element_rect(fill='transparent'), #transparent legend bg\n",
    "                         legend.box.background = element_rect(fill='white'), #transparent legend panel\n",
    "                         panel.border = element_rect(linetype=1, fill = NA),\n",
    "                       ))\n",
    "\n",
    "\n",
    "save(rs4633_plot, file = \"rs4633_plot.rdata\")\n",
    "rs4633_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAADAFBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////isF19AAAdoElEQVR4nO2de2AUxf3AhyCigCitP9Rqpf1Vfz778kpr0Zq0UioVsVqEglIEhaCiUDEiEEClNSCKVsQsiDya8DARSCKFkBAgISQkbLg8IC8SIHfJ5XXJqqHGB735zezt5XmX27vdvd0dvp8/7jszu7eZ48Puzc7M7SAMMA3SuwKAtoBgxgHBjAOCGUeR4HFoH8bCjGsG/+oQxrvvvxK14c48YAgCF/xVR2rDGCp4+s0HT84a0oK3Ll8pCvbkAUMQmGDLq2OHrNlx68BrZpPMue+fo4Jvj8G4FeWSgiOi4M48YAACFDwsw1U9YGutdSPGrohN31LBT41quvD296hat+DOPGAAAhT8LMaFl7ovv6vHYlHwfx5B/a8tpCVuwZ15wAAEKHgNxhfChz2+7StccW2tW/DK2/afmD2iFXsEd+YBAxCg4A/Jy38zXr7R8k1cv/79+6OwKV9dmkIu19evxZLgLnnAAAQhmOBEhZ+VlJQUIs5+vt+eboK75AEDELjg/BVFte9dLl6CxUt0+J0HS+YMrMCt1k0ox/plRx4wBIELLh/93cvvShWzouC6KcMHjyQxDlGsHXnAEEBXJeOAYMYBwYwDghkHBDMOCGYcEMw4IJhxAhL8eStgHr4MXPBDywDTsPTlwAXPEwDT4ATBbAOCGQcEMw4IZhwQzDggmHFAMOOAYMYBwYwDghkHBDMOCGYcEMw4IJhxfAu+EDXpgDuVvHSJw/MKgk2Gb8Gulh1uwc4oV+Ub0isINht9XaIlwdlbMI6UXjEuSEh4tvdhPvkt4QHCe5s3b36NxLHjCA6y5bVxbp6iu02XMstJ2i6lx8WRTLYnc4Rk4jwZO8ksl9LT6fufljKvknS9Z6/Nofq3MiUyBKcmYjxTesX4n6NHP937MHEWiZUcx0V5MnVkyzwpPZ7u9qCUeYmkbZ69Ykkmw5M5QDKcJ1NDMi9J6Qfp+x+WMvQa4vDs9X6o/q1MScBnsPdLdPl2wl5CflFRUc7eT5cs2ZeRkeEkWwoy3Bylux2RMidIullKZ5SRjM2TsdGDeTLNJHNCStNTWzgqZQpIuqXz/dZxk0L172U6/Al2CbglylURI73K/A5u5Dhqqslut9OzWKgjiSb610ispQWOLgV2xR8izxKh+Bis0ofgFTOfXYfPzybt5+hoh+dVtuBCEnhyvY6n+XiSOE5iMYkf0YIEksghsZxErpX8V2hsUfAhQLBvtLkPbonj8sipeZTY+xc9R/9FEtkk5pO4gRZsJ4lDJFqpYBvN57cG/yFAsG+0EdzKB0gcl3Y6+A8Bgn2jUU9WZUlgxHNp5cF/CBDsG4N0VSZwaSVnguaAJVx5S41RDCK4pvBYoFf1Luy03Mc3qF8pJjCIYKG5ojx4sp59nq/XoFIsYBTBCrGCYB+AYMYxiOD8A2InZdDEri5VqyqMYRDBn3BcmpL3jxZHKYDeGERw1SfcPkFo/DeBdl6fIDGXxBZaQMeUiknMojvuI4lqEktJPEgL0kmiAgT7xCCChWOJFYJQTzsuqc+DJKbQ2tGCSpLIJjGR7vchSZwk8TiJ22gB7QUtAME+MYrg5gr6kkOgN7TlJBaT2EoL6Cl9mkQr3e8YSdBv6zMk8rSAJ4mzwv2Wtz6tqFQFxrpMjCJYIeGWV7ksBV0lXbDq/VnUhRHBv7e8zhVWq0AFCDak4D9Y9lY71ThQPX+iLgiMexfOjGDSxj5MGtS0AXaOtr1p6VES6cSDWlpA5//kkUi/uRtoAZ0SWCA1153/lu6j64O8sNfp87n9w4jgitLG3tNEhCQSD5N4lhbQxttezn2/Xedprh8g8VMSq9xtdNKsO1UUBAV8jU4f3C+MCBaxkgZ1FYm1tO1NC4pIpK3zelpAr+AnSaRz/JpoAfkvIZSRWEL35BKU/OVTINjgKBZsLWlWqy7qAoJFSuM/VfL2CvItbND7Z0YEv7nspJ5/vrW+CARritiKVkZxrZJ3nwTBmqJccM3WFCVvB8HaolxwAbdDySkMgrVFueATHPeRgl9XgGBtUS64NmGrvbWhNljHIFhbVGhkUQ55OrQCBgRri2qCPwzynSBYWyaOUeXft0YUXL2RQDumUkk8RKKdFtDhhAwSaV92Iy04SxLZJO6h7wTB2nJa0ZS9DurXZwh05IFABe8hMZ1EGy2gbew0EveS2EALzpBEJolJZIdqKwg2BQ46huikI/90NMpOYp2ngDa/akmkIltoAR29cJBoE4Td3AEQrDEvzCzR74+DYO0Jt3BJBHrC8UluxK6pQ1JGnGObImXo0wYcUjqJjh9WfKpo3j0I1p4I9yN36ASOf0iP37mblk+TMlNpZpSUoY9xKvY8pWc7yey0eHk2lHxAsPa8PVmEno7r3MnJotNoKbOYZv4qZTiSLpPSk2mraZdlipI/3txYDIINjULBcJtkdBQLTt2p64C0b0CwiGLBW7k8dWqiNiBYBASD4L4BwcZGseA9HxepUxO1AcEitrxiZQeAVjTjnORrWpQ8blMzQLA6nKQ/UDLizxtAsDpUHDvO85V618ILIFikYJXChQESOWsVCDYuilvRiVwRCDYw6gguKm9UpzoqAoJF1BB8ljSzqtSpjoqAYJFdlnsmT95BEsc8g4j0B8aJngy9x5XGIyeLA8dzpcwKkm6giYlPVbScO8kreKq5RoBgEXFZnw+ELuv70Ml2Hev7nBM61/cZS/f3rO/zgtCxvs8uQTgNgg1LRlJSEu3MOuOZyEPnTBZ7MvSrNUdKp9Ld06QMXSzISRMPuwUbr5kFgtXhCbdgXsHCBNoAgtXh/WVHBMHG84YbcgDB6kBa0eS1DgSziluwAwSzCghmnC2r8wUQzDBiKxoEswsIZhwQzDgewQXutRiN06MFgtXhIcs/bILQ2PF4YQWL5aoLCFaHhyxRtJO6ni6ObK8BwczxF0vU3o5MMwhmDvd3sAQR7GyRh+b/EUCwOhzLONOZaZa/FECh1oZBsGoU5XqW5mgtkm9Y6+eIg2DVSOIyPMlWmRdoJwg2EXuDWF8TBJuFxNiizKAE26WVl5o0qBQFBKsDaUWXZgb+C8UuX8YaGQbB6vCEZW1eXh6dFu3Ik6Bzbc94MlRfmZQWF9UsFJOpyWlFRUWFycm7Dji0qRgIVocnxKmzqxoEIdsz15YuvrbJk6HzaRdL6dH0DROlzGySdpL4ywJtKgaC1WHtbyMiIn4z+6Ag5ERI0G/keE+GntuvS+mH6RuelDJ/I2lnRMRIS6o2FQPBKnKQS6CX4uLMzEw6bOgkMZNeeUtJFE9QWkCfmVhBIp0BImSTBJ1VL4wBwSbgMMfRZ2X2eujwYfdDhwWBFtCZ07kkikutbSQJ0jRz2EGwGajLSKe/gbCmp6fTE7aZxHRqvJhE8SE8tICewWUkiqsrHiSJakHYxo0GwSwDghlnG7dtZ7U2hwbBRmAbd4iH+2CGcdhPgGC2KQTBbPPK89naHBgEG4MIy8faHBgEGwMQzDggmHF+13VWppr0ITh56RIHjRULF86Nwu2TFi60gmCtGGNZtFuTA/sW7IxyVb4hpRN34/aFcAZryFhpzSfV8S04ewvGkVJ6bgtun7LgnTYQrBUfRs4tdmpxYN+CUxMxnulO2hZj7GrDyR+Q9DMITdOiIhc5OnR0dDmD4/aL4bP55KW5uvoFbapyUbOJU7imgC98C26JclXEYJdAks+Sa3O7C2e/BZdordBBME6Ojnbg87NJM/rvJFfy4qLoRhCsFZu4eS+e0uLAcB9sDDZx4ZZjWhwYBBuDI4ciQDDLFPIgmGlAMOOAYMbZvwcaWUwDrWjG2cQ9Ok6Tn5+BYGOgR08WCA4hH8eVanNgEGwMYNos44BgxinkJ0Iji2Wgo4Nx4jbAfTDTQEcH44BgxgHBjHM4HRpZTFPIwxnMNCf552dZtTgwCDYGztoTfP1RutT0YZUPDIINQRGfx9fPFZ9euU/VA4NgYxDHZfL1KydPnnyvZbuqBwbBxiCOy+DFJR9mgGAmiedSQTDLFCQddAvOy6hS9cAg2ChY+Xr/OwUOCDYKIJhxtsdpsibtRST4o5svvSnWk2maP2LgDXMaBGEZotTqWTE3G7gyGnbFqjux4+IRnBy2IGdZvx1S7qUhmwvir3qGCL4pi9Cia9VEJMHQivbJmCemXjd0WoOvzaPvpfuMknL3PEJeHv8FEXxnSCrnHxDsjzH932wpu+0ZKbewv8Q7UsEVS8nLyoHSo2zmX5Mj5N+wmAgeeMWw+9JDX9tepO45S8MMy/3jxo07TlIbxknQhT6WSGm6Co/whJSJIekzUvqh9d4Py5LgH5OX+IF17lxljsRZd96B1pDXjei0O+uc1+8S9BxJJL6f9vH9lwS+KJ3qSK3ol8X+6KMk9a5nzSW6DsRzUvpRus8YKbOApKul9K/neD8sS4KnkJci5OOxvA70Pnn9CEndCOuGrzu64eql7kzzzY+GooJ9Y+Xz6cIdDYczCLTRdypDgjYQeClNzQtZUoY+Oa1RSuf7OCxLgieQl+PoqDu3AEmslja7L9GXSpfoq5eQlxjPBfuxkaGtqjes/HbOrv5hWRI8gvxXf99zia7IkvAsdtGtkdU6mOpeMdC9uGvz/z0S6sr2xsqv52rUP2wPwaVmFjxkBv/JNZG+NnfcJsXeQ3ITvru5IG74w4IwbUNawv2XaLSmTSCERjAaucZpWsHTZ185ZKrv/r6Pbh7wI9rRsbg/ebE/c+PAG2adE4Qnbxw4/H4jtKKVCU4b95z3DT0Ev3ADGvCnXV+bU/DTetdAEVbe3tga9LtTLJO8b+j5Hew6Nhqh61JAcMix8jXiUuA+e2r6RK7gptU/RoNm/GY4CA45Vs9S4HXBvFuu4AHo1ncFnNl3o9qggk3O6aKijAPWogI+qC9iuYInZPSpFgRrSep6jhdOaSt4fjHGVbEgWBeObeZytRaMEjGO89frAYI1Yq/WgjeFo9vDw783FATrw7nyWqEgLyjBOS++6X1DN8Hu2Q1XrQDButHKrTvRRGlW6YA9LtEj9vqRC4K1pZXjUt13SzZ1DthNcOTuSBEQrB8fewSrND26m2AkjbGBYP3YGy8+0/B0gIKbbD76R7oJLmsqEwHB+rI/IWFbKhF8JoFAh/sPkphLop0W0L7MIyRmkdhAC+gosrxWdGQkXKKNQCLHcSlEcCWJHJ2TkELiQRJraAF9JF4qiXSIs54WnBNaGnfLEuyZBAGC9aW8uPjoESK4oZhAR5hOk0jnljXRAmq8ikR6DXfSgiahjHtVluCyMrhEG4TAvoPlCoZWtGE4zReWyqZBrmBoRRuGaj4AKmVfoqEVbRRabTVyKeUrm6o/ljnY8O3GqVM3XwDBZqKKr5Q/mvQ0vULDd7CpEAUXx+7wvrWH4KF/LC194EoQbCZEwT7pITh8NcZv/w4Emwki2JaZ62trj56s8YMmTRz0MAg2E1X8KZ6L97UVerJMD7mjyuBiM3K8b4WeLNPTWFqaxS2RO/FdDiDYaJRufUOm4MJ7rxw4cCAINhkOfpNMwb/+TtjNYbeAYJMhX/CgXUMqU8aDYJMhX/Dl6Te8/uYVINhkyBd8x7tPIDQKBJuMQu4Vy3jvm3oIbv28bdXfG0Gwyagkgh/wvglGk5ig+vD6bd63wGgSG9j5Eu8bYDSJDex8etxebxtgNIkN7DxnmeZtA4wmsUHZnjdlCIbRJNOSzy2WIRhGk0yLPMEEW3y83Y9fEGxA5AreNwihwWkg2HRY416VJfin/7t69Q9+BoJNh7xWNMaXb8R4/SAQbDrkCr5l5O7dlltBsOmw88dLvf5crYfgf/VDKGwrCDYdcrsqcf7y5bwfvyDYgJzYlOB9Q3fB345Y588uCDYk+b5mRvc4g+9eCoJNST63ahnnbUMPwWP7P9jxA/DkpUscNLZPWrjQ2pkFwUZEbkdHl75oZ5Sr8g1R8MJuWRBsRCr3vSVLcJe+6OwtGIvncvuUBe+0dWZBsBGRex/cZcpOaiLGM2nC1YaTP+jIbp01a1aoaw/4Ra7gLlN2upyy+LP5Hdn9K1bMDnXtAb/IFTx0+umTY9xTdlqiXBUx2CXgdhfOfkvKwiXaoJxOXy1L8KgkjFdJU3aSo6Md+PxsXPLiouhGKQuCDYrcVvQfhj0+4bKH/TwpCwQbj3xuSYTXr06vt0l+Ju2AYOMhtydL1qQdEGw8+A3yJr7LAgQbDzqatOdtLxtAMBsQwdY/Wmp7bwDBbGDnC1MtlureG+CB4GxQy/MyBMPEd9PSWn3qkH/BMPHdxNiyLJZzvYvhO5gVqGAvay3BY5RYQZ5geIySaZEnGB6jZFpsR19ZVt+7GB6jxAq2PI5r7F0Mj1FiBXmC4TFKpkWe4BN+3IJgw2LLXbna/3cwuv2NGhBsSuS1oqN+iPrdtx4EmxB5gjHmZ/WDvmgzIk9w07rfX4JGgGATIk9wf/SdWVkuEGxCbDJGkzB+LOlrP3ZBsEGRIxiWlzUxcgTD8rImRo5gWF7WxNjyNn4CPVkMY+NPeiuGnixWkCcYerJMizzB0JNlWmxHF73uf7ABerJMi7yODujJMi3Qk8U4sgTDk+7Mi+2wxXKmdzE86Y4V5I0mdX3SHQg2FfIEQ1+0aaGCN8S39Cz2/aQ7EGwuqOB/cv4E7xMBwSbEdjyvMtavYLhEmxYbf9LB+RUcExPz8rWTQLAJIYKdPN/as9jL74N3/BwEmxB5gw0vLVgw/46hINiE2I4+86Kjd7GX7+CwJSDYhMi7D87Nzc1v8OMXBBsSKji32P93cPZ7GSDYjMi5Dx57J04h1+g3QLAJkSN4+N/wH/utuONGEGxC5Age8PcLQ0fiZfCUHTNCBcdu7Fvwj26diZbgGTeAYBMipxW95RJ0Xd1Xw/8Cgk2ILfuBcfbexd1b0fasL/D53DoQbELkTpuVAQg2IlSwc+PG1B7FIJgVRMEct75HMQhmBfESXQKCmUVsZDXm5PYoBsGsIPcpOyDYpIBgxgHBjCMKdpaX9ygGwawgCq6BVjSzgGDGoYLLqkEws9iPzpqVfRQEM0vzqZJiefOiQbBZaQbBbAOCmabVduY4CGYYu8WSlbsrqUcpCGYGKjgbWtHsAoIZBwQzDhV87HBmj1IQzAxUMLSiGcYt2NnjN8IgmBncgivW13QrBcHM0Bi79hjfuo0r61YKghmCdlWCYIYBwYxDBe9JqOpWBoKZwZmRlge3SQwD98GMA4IZRz/Be5apRkyzSv8aDOIWnLCx+9T3UAievDZBLe6u8v/nLlbcgvW4TZp8LpCj98kEEOwTEMw4IJhx3ILLiruvbweCmSHgjo7kpUscNFYtWLSoEbdPWrjQCoINTWDzop1RrkrxsaSfteMja3H7wo4tQQo+/+KNA66d6ONhxcs2gWDFBCY4ewvGniWysjncPmXBO20kWZaePifAv+sW7ApHdy2dc/1x7+ZQOAhWDBVcxNd1K/MtODUR45nu5Pm59djVhpM/IOkYi2VGgH/XLfgA+um3GLe34W23DborE5ehsRFXPIXx1lsH/SwNj0AIzf/zgCb8PCorQw/85srIC9J+IFgugQ74d57B30Tni/Gz+e4twV2iV6Kl+Mvm5i/5sMdS7h4mlKHBm8PR8eNhv/joh4Pse9HticUZaLXr+yOJ+mFJ09EmaT8QLJdAb5NaolwVMdglYNfKdJJtd+Hst5QJXoZXIRSzHPXv3w8dKkMT8TsoYTnajVejePcl+tYfH0NriODH8Wk0VdoPBMuFCE4K6D44OTragc/PxtkTo6O34JIXF0U3KhGcgX5+oeY1Kvi9kpLiz8tQJF6DtncT/C4aPcBJBE/AJ9Ffpf1AsFyI4FF86+74ym6lobsPdkWgXyz5M4rhwywp3F1tkmCSW/+DwXZ81fXxVVgYhP5EmnHo8rf+gDZL+4FgubRMpIJ7loawo6Pthe9dOmLaKbz9jsuum/SNJBhvveXyn+3H+B9XoFiMZ6BdVPD48CtnXZD2A8Hy0ev3wbI7OjLvu/ZrKtj3AsYguC8MLxj9TxIGwcFjeMH+AcF9QQXr18gCwZqj17xoEBwiQDDjgGDGCWywQW3BvctBsMro24ruXQ6CVQYEMw4IZhwqWL9Jd73LQbDK6NuK7l3uPppnqlafU7ZAsBwMKdgzVavvKVsgWA4dgh3l5eVSj2UIBbdWd6VGEuyZqtUxZQsEBw0VvDWO/BOd4Dhui7sslIL5rpRIgj1TtTwRBAcPFZxDF5ktjo+PT3SXhVJwZVfOdAh2T9XyRBAcPD1ukyqKaw3wHeyZquWJIDh4egjeyVkNINgzVcsTQXDwGFJwx1QtTwTBQWNMwQEBgvuih+DkjUUgmCmgL5pxQDDjgGDGAcGM00NwWsIpEMwUcJvEOCCYcYwpuG3u9ZfevF7eeD8I7pMeggtzzhlAsCsc/XLZ1AnyxvtBcJ/o3IpuLSc0kbydxGqhc8D/AsZfyRvvB8F9orPgZo5ABWWRuFPoMuCPO8f9QbACdBbsjCecIfljJO4Vugz4485xfxCsgB6Cq8vrDfAdnCFeotvljfeD4D4xZCvaFYFGRpNGlqzxfhDcJ4YULA7037Re3ng/CO4TYwoOCBDcFyCYcWA0iXFAMOO01tX1KgPBjAOC2eXQ7jIQzDLQimYcEMw4BhEsDfhLAQSrB595xgCCOwf8aQDBaqO74M4BfxpAsNqEULDDIrGGlGVO7T3gLwMQHCgGELysSwDBKlJT3RBSwS2lEjZS1ljVc8BfDCBYRXZy+8v0/w7uHPCnAQSryE6O26O/4K4D/jfBbZKalCTtzjaA4IAAwYECghkHBDMOCGYcEMw4sCgH44BgxgHBjAOCGSckgrOq1WIsCA6QUAje+KRqzG7Q5F+BYUIhGNAREMw4IJhxQDDjgGDGAcGMA4IZBwQzDghmHBDMOCCYcUAw44BgxgHBjAOCGQcEMw4IZhwQzDggmHFAMOOAYMYBwYzTh+DkpUscXRIdWRBsJnwLdka5Kt/oTHRkQbCp8C04ewvGkZ2JjiwINhW+BacmYjyzM+HJzh827Em9Kw3IJ+AzuLm6+gW9Kw3Ix7fglihXRQx2CVLC/QqXaLPRVys6OtqBz8+WEtIrCDYZcB/MOCCYcUAw44BgxgHBjAOCGQcEMw4IZhwQzDggmHFAMOOAYMYBwYwDghkHBDNOMIKnzOvF03N6l4WWZ2fpXYN50+fqXYPIZ3qXxQQu2As/OaTwAIqJnah3DXBYs941mPOary0gWAVYFvxahcIDKCZznd41wJFtetdg+x5fW5QKBgwOCGYchYI7f7ekE1ULFi1q1LUGhLPjq/WtQHl0dKKPTcoEd/ndkk581o6PrNW1BoSVr+gr+OuoL31uUya4y++W9COb07kCJ+Nj9BVc+NrK5Wd8bFMmuONnTDpyfm69zjV47T86Cz74VLtjno9tpj+Dv4nO17cCODcR6yz4+Aqi4b/etykT3OV3SzrhWpmu698nbFu0bGrU53rW4Iu/ub54zsc2pa3ojt8t6UT2xOjoLbrWgKLzGYzTo18p9LEJ7oMZBwQzDghmHBDMOCCYcZgXPBhR7ilD3u/Xv7w2yvVLdBg3Dxvqq7+k7aql2lVPc5gX/PriUeiRxeubFuz2unktqsLHw35yIRKt7rXtWynOGXpewwpqDPOCMV6AtmNMzuAS9Oh9QzcsHPyrRrzttkF3ZYobf30neYlEkWF3fusubP3JkMGjCsn+Y+8d4RwzePBdTpyOtun7CZRwUQm+bPnwsN9OR8v5sMdS7h4mkG1fXTKNvLZcjVCWVPj56v27vnMv2b//0tg16P09i51YQM/o/BEUcFEJnoAfQSeq0IzlqH//fugQ2WZD4uTSf6JHsFTY9sT3h4RdSfZ/EOMUdMdTO1wYXzpe30+ghItKcCSehOxn0bTl6L2SkmLafdyE5tJddpMgFa5Cr1fePBC7G2VpURaUiP/bT/95fUFzUQrmwywp3F3iTLmrx9FXKlgqXI5WfRomCU59a+9S9AE+jV7V9xMo4aIUjLffcdl1k76hG6cOo21lKlgqbLj7ivG3SIL3337Z0D9/gTcgvUckFXARCO6THPSp331+NzIEFdGKi10wXhHnb4/z83JDURGNuOgFsw4IZhwQzDggmHFAMOP8PweI7k19kkxvAAAAAElFTkSuQmCC",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "surv_object_LIDs_rs4818 <- Surv(time = lids$Disease_duration_fromBL, event = lids$dyskinesia_status_at_censoring)\n",
    "\n",
    "fit_rs4818 <- survfit(surv_object_LIDs_rs4818 ~ lids$`chr22:19963684:C:G_G`, data = lids)\n",
    "\n",
    "rs4818_plot<-ggsurvplot(fit_rs4818, \n",
    "                        data = lids, \n",
    "                        pval = TRUE,\n",
    "                        title=\"rs4818\",\n",
    "                        linetype = \"strata\",\n",
    "                        censor=FALSE,\n",
    "                        palette = \"grey\",\n",
    "                        legend.title = \"Genotype   \",\n",
    "                        pval.coord = c(2, 0.65),\n",
    "                        legend=c(0.2,0.2),      \n",
    "                        legend.labs = c(\"GG\",\"CG\",\"CC\"),\n",
    "                        xlab= \"Time (Years)\",\n",
    "                        font.x=c(10,\"bold\",\"black\"),\n",
    "                        font.y=c(10,\"bold\",\"black\"),\n",
    "                        font.legend = c(10, \"bold\"), \n",
    "                        font.caption=c(10,\"bold\",\"black\"),\n",
    "                        ggtheme = theme(\n",
    "                         panel.background = element_rect(fill='#efefef'), #transparent panel bg\n",
    "                         plot.background = element_rect(fill='transparent', color=NA), #transparent plot bg\n",
    "                         panel.grid.major = element_blank(), #remove major gridlines\n",
    "                         panel.grid.minor = element_blank(), #remove minor gridlines\n",
    "                         legend.background = element_rect(fill='transparent'), #transparent legend bg\n",
    "                         legend.box.background = element_rect(fill='white'), #transparent legend panel\n",
    "                         panel.border = element_rect(linetype=1, fill = NA),\n",
    "                       ))\n",
    "\n",
    "\n",
    "save(rs4818_plot, file = \"rs4818_plot.rdata\")\n",
    "rs4818_plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "#install.packages('svglite')\n",
    "library(svglite)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "all<-cowplot::plot_grid(rs4633_plot$plot,\n",
    "                        rs6269_plot$plot,\n",
    "                        rs4680_plot$plot,\n",
    "                        rs4818_plot$plot,\n",
    "                        labels=c(\"A\",\"B\",\"C\",\"D\"))\n",
    "all\n",
    "ggsave(file=\"lids.svg\", plot=all, width=14, height=16,dpi = 300)\n",
    "ggsave(file=\"lids.png\", plot=all, width=14, height=16,dpi = 300)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# Copy to bucket\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/*.rdata {WORKSPACE_BUCKET}/COMT_NBA_loop/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp -r {WORK_DIR}/lids.* {WORKSPACE_BUCKET}/COMT_NBA_loop/')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "356.188px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
